{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8a03860e-7cf7-4d83-94fb-5f291e242aeb",
   "metadata": {},
   "source": [
    "# 08_interesting_virus_host_pairs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e2dc225-7150-4b63-a79d-7c1e85b7040c",
   "metadata": {},
   "source": [
    "This document explores which virus-host pairs are the most interesting based on the strength of the connection, the taxonomy of the host (biogeochemical significance), and quality of the viral sequence/MAG."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9dc9f314-cb82-4312-a067-2010dd982c3a",
   "metadata": {},
   "source": [
    "## Load packages and data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "66e07241-e5ec-403d-be5a-2774d627630e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import sys\n",
    "import csv\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "import glob\n",
    "import seaborn as sns\n",
    "from collections import Counter\n",
    "\n",
    "pd.set_option('display.max_columns', None)\n",
    "\n",
    "sags = pd.read_csv(\"~/Documents/Bigelow/Virus_Project/OMZ_MH_Analysis/Data/sag_data/MPvsag_info_230818.csv\")\n",
    "# rename classification column\n",
    "sags.rename(columns={'classification_via_GTDBTk': 'classification'}, inplace=True) # rename classification column\n",
    "# split the classification into tax level columns and rename\n",
    "sags[['domain', 'phyla', 'class', 'order', 'family', 'genus', 'species']]=sags.classification.str.split(';', expand=True)\n",
    "sags['domain'] = sags['domain'].str.replace('d__', '') # remove the d__ in front of all observations\n",
    "sags['phyla'] = sags['phyla'].str.replace('p__', '')\n",
    "sags['class'] = sags['class'].str.replace('c__', '')\n",
    "sags['order'] = sags['order'].str.replace('o__', '')\n",
    "sags['family'] = sags['family'].str.replace('f__', '')\n",
    "sags['genus'] = sags['genus'].str.replace('g__', '')\n",
    "sags['species'] = sags['species'].str.replace('s__', '')\n",
    "\n",
    "vMAGs = pd.read_csv('~/Documents/Bigelow/Virus_Project/OMZ_MH_Analysis/Data/proximeta_viral_files/vMAG_associations.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ceb3c8a-7e00-41de-a1e2-391d02554389",
   "metadata": {},
   "source": [
    "## SAGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "ae79420a-a8b8-402d-9ef1-58bc9014e4b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sags)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1bb898ac-bca6-4a4f-8a95-ad85c95a69d5",
   "metadata": {},
   "source": [
    "### Criteria: Taxonomy"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4508dd91-568c-477b-9a1a-2fc2b5091c6b",
   "metadata": {},
   "source": [
    "**Orders with more SAGs than vMAGs:**\n",
    "+ Brocadiales\n",
    "+ HIMB59\n",
    "+ Marinisomatales- The most abundant and largest difference\n",
    "+ PS1\n",
    "+ Peligibacterales\n",
    "+ Poseidoniales\n",
    "+ SAR324\n",
    "+ UB11654"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c3526658-4b64-4880-828b-7b156b84d126",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sag_abund_list = ['Brocadiales','HIMB59','Marinisomatales','PS1','Peligibacterales','Poseidoniales','SAR324','UB11654']\n",
    "sag_abund = sags[sags['order'].isin(sag_abund_list)]\n",
    "len(sag_abund)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d62c9b99-9fa0-472c-b4d0-c3c6a47a1632",
   "metadata": {},
   "source": [
    "### Criteria: Length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "26e87093-ef59-4857-8537-cc762bd63202",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "long_sags = sag_abund[sag_abund['contig_length'] >= 15000]\n",
    "len(long_sags)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aced8dbb-669e-44ce-8b09-0628141a7827",
   "metadata": {},
   "source": [
    "### Criteria: Viral Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "587ae5c3-8450-41a8-9aaf-54038675fcb3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sag_viral_genes = long_sags[long_sags['viral_genes'] != 0]\n",
    "sag_viral_genes = sag_viral_genes.sort_values(by='viral_genes',ascending=False)\n",
    "len(sag_viral_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "597a2046-279d-4e1f-8b14-b1daf6975d55",
   "metadata": {},
   "source": [
    "### Results\n",
    "\n",
    "SAGs row 15: highest viral genes\n",
    "+ vir_id = vir_AM_656-P04\n",
    "+ sag = AM=656_P04\n",
    "+ depth = 95\n",
    "+ order = Marinisomatales\n",
    "\n",
    "SAGs row 41: second highest viral genes\n",
    "+ vir_id = vir_AM-666-P13\n",
    "+ sag = AM-666-P13\n",
    "+ depth = 400\n",
    "+ order = HIMB59\n",
    "\n",
    "SAGs row 26: highest completeness (100)\n",
    "+ vir_id = vir_AM-662-D22\n",
    "+ sag = AM-662-D22\n",
    "+ depth = 140\n",
    "+ order = PS1\n",
    "\n",
    "SAGs row 5: highest viral genes for 80 depth\n",
    "+ vir_id = vir_AM-654-B18\n",
    "+ sag = AM-654-B18\n",
    "+ depth = 80\n",
    "+ order = Brocadiales\n",
    "\n",
    "SAGs row 27: highest quality SAR324 (4 out of 9)\n",
    "+ vir_id = vir_AM-662-A03\n",
    "+ sag = AM-662-A03\n",
    "+ depth = 140\n",
    "+ order = SAR324"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b1ca1163-b097-4978-9404-c2c032da9e63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>vir_id</th>\n",
       "      <th>sag</th>\n",
       "      <th>contig_length</th>\n",
       "      <th>provirus</th>\n",
       "      <th>proviral_length</th>\n",
       "      <th>gene_count</th>\n",
       "      <th>viral_genes</th>\n",
       "      <th>host_genes</th>\n",
       "      <th>checkv_quality</th>\n",
       "      <th>miuvig_quality</th>\n",
       "      <th>completeness</th>\n",
       "      <th>completeness_method</th>\n",
       "      <th>contamination</th>\n",
       "      <th>kmer_freq</th>\n",
       "      <th>warnings</th>\n",
       "      <th>classification</th>\n",
       "      <th>plate</th>\n",
       "      <th>depth</th>\n",
       "      <th>domain</th>\n",
       "      <th>phyla</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>vir_AM-656-P04</td>\n",
       "      <td>AM-656-P04</td>\n",
       "      <td>39217</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>45</td>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>86.60</td>\n",
       "      <td>AAI-based (high-confidence)</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Marinisomatota;c__Marinisomatia...</td>\n",
       "      <td>AM-656</td>\n",
       "      <td>95</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Marinisomatota</td>\n",
       "      <td>Marinisomatia</td>\n",
       "      <td>Marinisomatales</td>\n",
       "      <td>TCS55</td>\n",
       "      <td>UBA2126</td>\n",
       "      <td>UBA2126 sp002730315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>vir_AM-666-P13</td>\n",
       "      <td>AM-666-P13</td>\n",
       "      <td>129323</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>178</td>\n",
       "      <td>25</td>\n",
       "      <td>19</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>47.53</td>\n",
       "      <td>HMM-based (lower-bound)</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>AM-666</td>\n",
       "      <td>400</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>HIMB59</td>\n",
       "      <td>GCA-002718135</td>\n",
       "      <td>MarineAlpha5-Bin3</td>\n",
       "      <td>MarineAlpha5-Bin3 sp002938255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>vir_AM-656-A07</td>\n",
       "      <td>AM-656-A07</td>\n",
       "      <td>83878</td>\n",
       "      <td>Yes</td>\n",
       "      <td>54319.0</td>\n",
       "      <td>86</td>\n",
       "      <td>24</td>\n",
       "      <td>30</td>\n",
       "      <td>Complete</td>\n",
       "      <td>High-quality</td>\n",
       "      <td>100.00</td>\n",
       "      <td>Provirus (high-confidence)</td>\n",
       "      <td>35.24</td>\n",
       "      <td>1.0</td>\n",
       "      <td>contig &gt;1.5x longer than expected genome length</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>AM-656</td>\n",
       "      <td>95</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>HIMB59</td>\n",
       "      <td>GCA-002718135</td>\n",
       "      <td>JAGFWP01</td>\n",
       "      <td>JAGFWP01 sp017639935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>vir_AM-662-D22</td>\n",
       "      <td>AM-662-D22</td>\n",
       "      <td>40167</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>57</td>\n",
       "      <td>19</td>\n",
       "      <td>1</td>\n",
       "      <td>High-quality</td>\n",
       "      <td>High-quality</td>\n",
       "      <td>100.00</td>\n",
       "      <td>AAI-based (high-confidence)</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>AM-662</td>\n",
       "      <td>140</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>PS1</td>\n",
       "      <td>Thioglobaceae</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>vir_AM-662-A03</td>\n",
       "      <td>AM-662-A03</td>\n",
       "      <td>131913</td>\n",
       "      <td>Yes</td>\n",
       "      <td>129050.0</td>\n",
       "      <td>153</td>\n",
       "      <td>16</td>\n",
       "      <td>11</td>\n",
       "      <td>High-quality</td>\n",
       "      <td>High-quality</td>\n",
       "      <td>91.08</td>\n",
       "      <td>HMM-based (lower-bound)</td>\n",
       "      <td>2.17</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...</td>\n",
       "      <td>AM-662</td>\n",
       "      <td>140</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>NAC60-12</td>\n",
       "      <td>UBA1014</td>\n",
       "      <td>UBA1014 sp002683785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>vir_AM-654-B18</td>\n",
       "      <td>AM-654-B18</td>\n",
       "      <td>25490</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>50.74</td>\n",
       "      <td>AAI-based (high-confidence)</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Planctomycetota;c__Brocadiae;o_...</td>\n",
       "      <td>AM-654</td>\n",
       "      <td>80</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Planctomycetota</td>\n",
       "      <td>Brocadiae</td>\n",
       "      <td>Brocadiales</td>\n",
       "      <td>Scalinduaceae</td>\n",
       "      <td>Scalindua</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>vir_AM-656-P20</td>\n",
       "      <td>AM-656-P20</td>\n",
       "      <td>33778</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>45</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>50.05</td>\n",
       "      <td>HMM-based (lower-bound)</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...</td>\n",
       "      <td>AM-656</td>\n",
       "      <td>95</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>NAC60-12</td>\n",
       "      <td>UBA1014</td>\n",
       "      <td>UBA1014 sp002683785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>vir_AM-654-K16</td>\n",
       "      <td>AM-654-K16</td>\n",
       "      <td>69480</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>82</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>43.45</td>\n",
       "      <td>HMM-based (lower-bound)</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...</td>\n",
       "      <td>AM-654</td>\n",
       "      <td>80</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>NAC60-12</td>\n",
       "      <td>UBA1014</td>\n",
       "      <td>UBA1014 sp002683785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>vir_AM-654-I17</td>\n",
       "      <td>AM-654-I17</td>\n",
       "      <td>27722</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>15.02</td>\n",
       "      <td>HMM-based (lower-bound)</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...</td>\n",
       "      <td>AM-654</td>\n",
       "      <td>80</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>SAR324</td>\n",
       "      <td>NAC60-12</td>\n",
       "      <td>Arctic96AD-7</td>\n",
       "      <td>Arctic96AD-7 sp002685535</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            vir_id         sag  contig_length provirus  proviral_length  \\\n",
       "15  vir_AM-656-P04  AM-656-P04          39217       No              NaN   \n",
       "41  vir_AM-666-P13  AM-666-P13         129323       No              NaN   \n",
       "16  vir_AM-656-A07  AM-656-A07          83878      Yes          54319.0   \n",
       "26  vir_AM-662-D22  AM-662-D22          40167       No              NaN   \n",
       "27  vir_AM-662-A03  AM-662-A03         131913      Yes         129050.0   \n",
       "5   vir_AM-654-B18  AM-654-B18          25490       No              NaN   \n",
       "6   vir_AM-656-P20  AM-656-P20          33778       No              NaN   \n",
       "23  vir_AM-654-K16  AM-654-K16          69480       No              NaN   \n",
       "20  vir_AM-654-I17  AM-654-I17          27722       No              NaN   \n",
       "\n",
       "    gene_count  viral_genes  host_genes  checkv_quality   miuvig_quality  \\\n",
       "15          45           28           1  Medium-quality  Genome-fragment   \n",
       "41         178           25          19     Low-quality  Genome-fragment   \n",
       "16          86           24          30        Complete     High-quality   \n",
       "26          57           19           1    High-quality     High-quality   \n",
       "27         153           16          11    High-quality     High-quality   \n",
       "5           25            4           0  Medium-quality  Genome-fragment   \n",
       "6           45            4           3  Medium-quality  Genome-fragment   \n",
       "23          82            3           2     Low-quality  Genome-fragment   \n",
       "20          26            1           5     Low-quality  Genome-fragment   \n",
       "\n",
       "    completeness          completeness_method  contamination  kmer_freq  \\\n",
       "15         86.60  AAI-based (high-confidence)           0.00        1.0   \n",
       "41         47.53      HMM-based (lower-bound)           0.00        1.0   \n",
       "16        100.00   Provirus (high-confidence)          35.24        1.0   \n",
       "26        100.00  AAI-based (high-confidence)           0.00        1.0   \n",
       "27         91.08      HMM-based (lower-bound)           2.17        1.0   \n",
       "5          50.74  AAI-based (high-confidence)           0.00        1.0   \n",
       "6          50.05      HMM-based (lower-bound)           0.00        1.0   \n",
       "23         43.45      HMM-based (lower-bound)           0.00        1.0   \n",
       "20         15.02      HMM-based (lower-bound)           0.00        1.0   \n",
       "\n",
       "                                           warnings  \\\n",
       "15                                              NaN   \n",
       "41                                              NaN   \n",
       "16  contig >1.5x longer than expected genome length   \n",
       "26                                              NaN   \n",
       "27                                              NaN   \n",
       "5                                               NaN   \n",
       "6                                               NaN   \n",
       "23                                              NaN   \n",
       "20                                              NaN   \n",
       "\n",
       "                                       classification   plate  depth  \\\n",
       "15  d__Bacteria;p__Marinisomatota;c__Marinisomatia...  AM-656     95   \n",
       "41  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  AM-666    400   \n",
       "16  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  AM-656     95   \n",
       "26  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  AM-662    140   \n",
       "27  d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...  AM-662    140   \n",
       "5   d__Bacteria;p__Planctomycetota;c__Brocadiae;o_...  AM-654     80   \n",
       "6   d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...  AM-656     95   \n",
       "23  d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...  AM-654     80   \n",
       "20  d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...  AM-654     80   \n",
       "\n",
       "      domain            phyla                class            order  \\\n",
       "15  Bacteria   Marinisomatota        Marinisomatia  Marinisomatales   \n",
       "41  Bacteria   Proteobacteria  Alphaproteobacteria           HIMB59   \n",
       "16  Bacteria   Proteobacteria  Alphaproteobacteria           HIMB59   \n",
       "26  Bacteria   Proteobacteria  Gammaproteobacteria              PS1   \n",
       "27  Bacteria           SAR324               SAR324           SAR324   \n",
       "5   Bacteria  Planctomycetota            Brocadiae      Brocadiales   \n",
       "6   Bacteria           SAR324               SAR324           SAR324   \n",
       "23  Bacteria           SAR324               SAR324           SAR324   \n",
       "20  Bacteria           SAR324               SAR324           SAR324   \n",
       "\n",
       "           family              genus                        species  \n",
       "15          TCS55            UBA2126            UBA2126 sp002730315  \n",
       "41  GCA-002718135  MarineAlpha5-Bin3  MarineAlpha5-Bin3 sp002938255  \n",
       "16  GCA-002718135           JAGFWP01           JAGFWP01 sp017639935  \n",
       "26  Thioglobaceae                                                    \n",
       "27       NAC60-12            UBA1014            UBA1014 sp002683785  \n",
       "5   Scalinduaceae          Scalindua                                 \n",
       "6        NAC60-12            UBA1014            UBA1014 sp002683785  \n",
       "23       NAC60-12            UBA1014            UBA1014 sp002683785  \n",
       "20       NAC60-12       Arctic96AD-7       Arctic96AD-7 sp002685535  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sag_viral_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1be767b-980e-4ad3-803d-e5470955ec26",
   "metadata": {},
   "source": [
    "## vMAGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "2b3460dc-4a1d-47c6-b473-7919b0e3f9fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "132"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(vMAGs)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7fda7450-c662-4a46-b582-12f5e54b5ab0",
   "metadata": {},
   "source": [
    "## Criteria: Interesting Taxonomy"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d4cfae2-0f44-4d1f-8801-d278aec76d31",
   "metadata": {},
   "source": [
    "**Orders with more vMAGs than SAGs:**\n",
    "+ Arenicellales\n",
    "+ Dehalociccoidakes\n",
    "+ Flavobacteriales\n",
    "+ Hydrothermarchaeales\n",
    "+ JABGRK01\n",
    "+ Mitrosopherales\n",
    "+ Pedospherales\n",
    "+ Phycisphaerales\n",
    "+ Pirellules \n",
    "+ Pseudomonadales\n",
    "+ UBA1135 \n",
    "+ UBA1369\n",
    "+ UBA7887\n",
    "+ UBA890\n",
    "+ Verrucomicrobales\n",
    "+ Vicinaibacterales\n",
    "+ Wosearchaeales\n",
    "+ Woeseiales"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "4a94118b-adfa-451d-b586-4464dc1d78c6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Pseudomonadales', nan, 'Anaerolineales', 'Pirellulales',\n",
       "       'Verrucomicrobiales', 'Caulobacterales', 'PCC-6307', 'UBA4486',\n",
       "       'Brocadiales', 'Acidimicrobiales', 'GCA-2731375', 'UBA5377'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vMAGs['order'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "21993862-a1cf-497f-bb07-2d69ba89ed58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vMAG_abund_list = ['Arenicellales', 'Dehalociccoidakes', 'Flavobacteriales', 'Hydrothermarchaeales', 'JABGRK01', 'Mitrosopherales', 'Pedospherales', \n",
    "                  'Phycisphaerales', 'Pirellules', 'Pseudomonadales', 'UBA1135', 'UBA1369', 'UBA7887', 'UBA890', 'Verrucomicrobales', 'Vicinaibacterales',\n",
    "                  'Wosearchaeales', 'Woeseiales']\n",
    "vMAG_abund = vMAGs[vMAGs['order'].isin(vMAG_abund_list)]\n",
    "len(vMAG_abund)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "eccd2161-45bc-40ff-bbad-cd0d11acb8c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>virus_name</th>\n",
       "      <th>virus_length</th>\n",
       "      <th>virus_read_count</th>\n",
       "      <th>virus_read_depth</th>\n",
       "      <th>virus_read_depth_in_host</th>\n",
       "      <th>host_name</th>\n",
       "      <th>host_length</th>\n",
       "      <th>host_read_count</th>\n",
       "      <th>host_read_depth</th>\n",
       "      <th>intra_read_count</th>\n",
       "      <th>intra_linkage_density</th>\n",
       "      <th>inter_read_count</th>\n",
       "      <th>raw_inter_linkage_density</th>\n",
       "      <th>raw_inter_vs_intra_ratio</th>\n",
       "      <th>viral_copies_per_cell</th>\n",
       "      <th>adjusted_inter_linkage_density</th>\n",
       "      <th>adjusted_inter_vs_intra_ratio</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>virus_type</th>\n",
       "      <th>sample_depth</th>\n",
       "      <th>classification</th>\n",
       "      <th>fastani_reference</th>\n",
       "      <th>fastani_reference_radius</th>\n",
       "      <th>fastani_taxonomy</th>\n",
       "      <th>fastani_ani</th>\n",
       "      <th>fastani_af</th>\n",
       "      <th>closest_placement_reference</th>\n",
       "      <th>closest_placement_radius</th>\n",
       "      <th>closest_placement_taxonomy</th>\n",
       "      <th>closest_placement_ani</th>\n",
       "      <th>closest_placement_af</th>\n",
       "      <th>pplacer_taxonomy</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>note</th>\n",
       "      <th>other_related_references(genome_id,species_name,radius,ANI,AF)</th>\n",
       "      <th>msa_percent</th>\n",
       "      <th>translation_table</th>\n",
       "      <th>red_value</th>\n",
       "      <th>warnings_x</th>\n",
       "      <th>domain</th>\n",
       "      <th>phyla</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>contig_id</th>\n",
       "      <th>contig_length</th>\n",
       "      <th>provirus</th>\n",
       "      <th>proviral_length</th>\n",
       "      <th>gene_count</th>\n",
       "      <th>viral_genes</th>\n",
       "      <th>host_genes</th>\n",
       "      <th>checkv_quality</th>\n",
       "      <th>miuvig_quality</th>\n",
       "      <th>completeness</th>\n",
       "      <th>completeness_method</th>\n",
       "      <th>contamination</th>\n",
       "      <th>kmer_freq</th>\n",
       "      <th>warnings_y</th>\n",
       "      <th>N</th>\n",
       "      <th>L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>vMAG_32</td>\n",
       "      <td>21616</td>\n",
       "      <td>791</td>\n",
       "      <td>36.593264</td>\n",
       "      <td>0.741755</td>\n",
       "      <td>bin_23</td>\n",
       "      <td>3688193</td>\n",
       "      <td>23817</td>\n",
       "      <td>6.457634</td>\n",
       "      <td>357</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.703704</td>\n",
       "      <td>0.114865</td>\n",
       "      <td>0.000328</td>\n",
       "      <td>6.126361</td>\n",
       "      <td>JV119</td>\n",
       "      <td>vMAG</td>\n",
       "      <td>400</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification fully defined by topo...</td>\n",
       "      <td>classification based on placement in class-lev...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48.26</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.75999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>HTCC2089</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vMAG_32|N=2|L=21616</td>\n",
       "      <td>21816</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>5.53</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>21616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>vMAG_32</td>\n",
       "      <td>21616</td>\n",
       "      <td>791</td>\n",
       "      <td>36.593264</td>\n",
       "      <td>0.741755</td>\n",
       "      <td>bin_23</td>\n",
       "      <td>3688193</td>\n",
       "      <td>23817</td>\n",
       "      <td>6.457634</td>\n",
       "      <td>357</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.703704</td>\n",
       "      <td>0.114865</td>\n",
       "      <td>0.000328</td>\n",
       "      <td>6.126361</td>\n",
       "      <td>JV119</td>\n",
       "      <td>vMAG</td>\n",
       "      <td>400</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification fully defined by topo...</td>\n",
       "      <td>classification based on placement in class-lev...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48.26</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.75999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>HTCC2089</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vMAG_32|N=3|L=73423</td>\n",
       "      <td>73823</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>49</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>21.85</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>73423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>vMAG_32</td>\n",
       "      <td>21616</td>\n",
       "      <td>791</td>\n",
       "      <td>36.593264</td>\n",
       "      <td>0.741755</td>\n",
       "      <td>bin_23</td>\n",
       "      <td>3688193</td>\n",
       "      <td>23817</td>\n",
       "      <td>6.457634</td>\n",
       "      <td>357</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.703704</td>\n",
       "      <td>0.114865</td>\n",
       "      <td>0.000328</td>\n",
       "      <td>6.126361</td>\n",
       "      <td>JV119</td>\n",
       "      <td>vMAG</td>\n",
       "      <td>400</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification fully defined by topo...</td>\n",
       "      <td>classification based on placement in class-lev...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48.26</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.75999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>HTCC2089</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vMAG_32|N=2|L=59685</td>\n",
       "      <td>59885</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>32.90</td>\n",
       "      <td>AAI-based (high-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>59685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>vMAG_32</td>\n",
       "      <td>21616</td>\n",
       "      <td>791</td>\n",
       "      <td>36.593264</td>\n",
       "      <td>0.741755</td>\n",
       "      <td>bin_23</td>\n",
       "      <td>3688193</td>\n",
       "      <td>23817</td>\n",
       "      <td>6.457634</td>\n",
       "      <td>357</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.703704</td>\n",
       "      <td>0.114865</td>\n",
       "      <td>0.000328</td>\n",
       "      <td>6.126361</td>\n",
       "      <td>JV119</td>\n",
       "      <td>vMAG</td>\n",
       "      <td>400</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification fully defined by topo...</td>\n",
       "      <td>classification based on placement in class-lev...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48.26</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.75999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>HTCC2089</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vMAG_32|N=2|L=25053</td>\n",
       "      <td>25253</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>40.09</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>25053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>vMAG_11</td>\n",
       "      <td>12392</td>\n",
       "      <td>755</td>\n",
       "      <td>60.926404</td>\n",
       "      <td>1.249772</td>\n",
       "      <td>bin_26</td>\n",
       "      <td>2922201</td>\n",
       "      <td>11622</td>\n",
       "      <td>3.977139</td>\n",
       "      <td>1037</td>\n",
       "      <td>0.000250</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000110</td>\n",
       "      <td>0.441120</td>\n",
       "      <td>0.314239</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>1.403773</td>\n",
       "      <td>JV154</td>\n",
       "      <td>vMAG</td>\n",
       "      <td>140</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>GCA_002730035.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>99.8</td>\n",
       "      <td>0.964</td>\n",
       "      <td>GCA_002730035.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>99.8</td>\n",
       "      <td>0.964</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>topological placement and ANI have congruent s...</td>\n",
       "      <td>GCA_003451715.1, s__UBA9145 sp003451715, 95.0,...</td>\n",
       "      <td>73.72</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>Pseudohongiellaceae</td>\n",
       "      <td>UBA9145</td>\n",
       "      <td>UBA9145 sp002730035</td>\n",
       "      <td>vMAG_11|N=2|L=11750</td>\n",
       "      <td>11950</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Not-determined</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>no viral genes detected</td>\n",
       "      <td>2</td>\n",
       "      <td>11750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>vMAG_11</td>\n",
       "      <td>12392</td>\n",
       "      <td>755</td>\n",
       "      <td>60.926404</td>\n",
       "      <td>1.249772</td>\n",
       "      <td>bin_26</td>\n",
       "      <td>2922201</td>\n",
       "      <td>11622</td>\n",
       "      <td>3.977139</td>\n",
       "      <td>1037</td>\n",
       "      <td>0.000250</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000110</td>\n",
       "      <td>0.441120</td>\n",
       "      <td>0.314239</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>1.403773</td>\n",
       "      <td>JV154</td>\n",
       "      <td>vMAG</td>\n",
       "      <td>140</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>GCA_002730035.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>99.8</td>\n",
       "      <td>0.964</td>\n",
       "      <td>GCA_002730035.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>99.8</td>\n",
       "      <td>0.964</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>topological placement and ANI have congruent s...</td>\n",
       "      <td>GCA_003451715.1, s__UBA9145 sp003451715, 95.0,...</td>\n",
       "      <td>73.72</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>Pseudohongiellaceae</td>\n",
       "      <td>UBA9145</td>\n",
       "      <td>UBA9145 sp002730035</td>\n",
       "      <td>vMAG_11|N=2|L=53693</td>\n",
       "      <td>53893</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>High-quality</td>\n",
       "      <td>High-quality</td>\n",
       "      <td>100.00</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>53693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>vMAG_11</td>\n",
       "      <td>12392</td>\n",
       "      <td>755</td>\n",
       "      <td>60.926404</td>\n",
       "      <td>1.249772</td>\n",
       "      <td>bin_26</td>\n",
       "      <td>2922201</td>\n",
       "      <td>11622</td>\n",
       "      <td>3.977139</td>\n",
       "      <td>1037</td>\n",
       "      <td>0.000250</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000110</td>\n",
       "      <td>0.441120</td>\n",
       "      <td>0.314239</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>1.403773</td>\n",
       "      <td>JV154</td>\n",
       "      <td>vMAG</td>\n",
       "      <td>140</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>GCA_002730035.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>99.8</td>\n",
       "      <td>0.964</td>\n",
       "      <td>GCA_002730035.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>99.8</td>\n",
       "      <td>0.964</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>topological placement and ANI have congruent s...</td>\n",
       "      <td>GCA_003451715.1, s__UBA9145 sp003451715, 95.0,...</td>\n",
       "      <td>73.72</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>Pseudohongiellaceae</td>\n",
       "      <td>UBA9145</td>\n",
       "      <td>UBA9145 sp002730035</td>\n",
       "      <td>vMAG_11|N=2|L=48736</td>\n",
       "      <td>48936</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>52.93</td>\n",
       "      <td>AAI-based (high-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>48736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>vMAG_11</td>\n",
       "      <td>12392</td>\n",
       "      <td>755</td>\n",
       "      <td>60.926404</td>\n",
       "      <td>1.249772</td>\n",
       "      <td>bin_26</td>\n",
       "      <td>2922201</td>\n",
       "      <td>11622</td>\n",
       "      <td>3.977139</td>\n",
       "      <td>1037</td>\n",
       "      <td>0.000250</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000110</td>\n",
       "      <td>0.441120</td>\n",
       "      <td>0.314239</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>1.403773</td>\n",
       "      <td>JV154</td>\n",
       "      <td>vMAG</td>\n",
       "      <td>140</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>GCA_002730035.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>99.8</td>\n",
       "      <td>0.964</td>\n",
       "      <td>GCA_002730035.1</td>\n",
       "      <td>95.0</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>99.8</td>\n",
       "      <td>0.964</td>\n",
       "      <td>d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>topological placement and ANI have congruent s...</td>\n",
       "      <td>GCA_003451715.1, s__UBA9145 sp003451715, 95.0,...</td>\n",
       "      <td>73.72</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>Pseudohongiellaceae</td>\n",
       "      <td>UBA9145</td>\n",
       "      <td>UBA9145 sp002730035</td>\n",
       "      <td>vMAG_11|N=2|L=12392</td>\n",
       "      <td>12592</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>6.70</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>12392</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   virus_name  virus_length  virus_read_count  virus_read_depth  \\\n",
       "0     vMAG_32         21616               791         36.593264   \n",
       "1     vMAG_32         21616               791         36.593264   \n",
       "2     vMAG_32         21616               791         36.593264   \n",
       "3     vMAG_32         21616               791         36.593264   \n",
       "28    vMAG_11         12392               755         60.926404   \n",
       "29    vMAG_11         12392               755         60.926404   \n",
       "30    vMAG_11         12392               755         60.926404   \n",
       "31    vMAG_11         12392               755         60.926404   \n",
       "\n",
       "    virus_read_depth_in_host host_name  host_length  host_read_count  \\\n",
       "0                   0.741755    bin_23      3688193            23817   \n",
       "1                   0.741755    bin_23      3688193            23817   \n",
       "2                   0.741755    bin_23      3688193            23817   \n",
       "3                   0.741755    bin_23      3688193            23817   \n",
       "28                  1.249772    bin_26      2922201            11622   \n",
       "29                  1.249772    bin_26      2922201            11622   \n",
       "30                  1.249772    bin_26      2922201            11622   \n",
       "31                  1.249772    bin_26      2922201            11622   \n",
       "\n",
       "    host_read_depth  intra_read_count  intra_linkage_density  \\\n",
       "0          6.457634               357               0.000053   \n",
       "1          6.457634               357               0.000053   \n",
       "2          6.457634               357               0.000053   \n",
       "3          6.457634               357               0.000053   \n",
       "28         3.977139              1037               0.000250   \n",
       "29         3.977139              1037               0.000250   \n",
       "30         3.977139              1037               0.000250   \n",
       "31         3.977139              1037               0.000250   \n",
       "\n",
       "    inter_read_count  raw_inter_linkage_density  raw_inter_vs_intra_ratio  \\\n",
       "0                  3                   0.000038                  0.703704   \n",
       "1                  3                   0.000038                  0.703704   \n",
       "2                  3                   0.000038                  0.703704   \n",
       "3                  3                   0.000038                  0.703704   \n",
       "28                 4                   0.000110                  0.441120   \n",
       "29                 4                   0.000110                  0.441120   \n",
       "30                 4                   0.000110                  0.441120   \n",
       "31                 4                   0.000110                  0.441120   \n",
       "\n",
       "    viral_copies_per_cell  adjusted_inter_linkage_density  \\\n",
       "0                0.114865                        0.000328   \n",
       "1                0.114865                        0.000328   \n",
       "2                0.114865                        0.000328   \n",
       "3                0.114865                        0.000328   \n",
       "28               0.314239                        0.000352   \n",
       "29               0.314239                        0.000352   \n",
       "30               0.314239                        0.000352   \n",
       "31               0.314239                        0.000352   \n",
       "\n",
       "    adjusted_inter_vs_intra_ratio sample_name virus_type  sample_depth  \\\n",
       "0                        6.126361       JV119       vMAG           400   \n",
       "1                        6.126361       JV119       vMAG           400   \n",
       "2                        6.126361       JV119       vMAG           400   \n",
       "3                        6.126361       JV119       vMAG           400   \n",
       "28                       1.403773       JV154       vMAG           140   \n",
       "29                       1.403773       JV154       vMAG           140   \n",
       "30                       1.403773       JV154       vMAG           140   \n",
       "31                       1.403773       JV154       vMAG           140   \n",
       "\n",
       "                                       classification fastani_reference  \\\n",
       "0   d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...               NaN   \n",
       "1   d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...               NaN   \n",
       "2   d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...               NaN   \n",
       "3   d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...               NaN   \n",
       "28  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   GCA_002730035.1   \n",
       "29  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   GCA_002730035.1   \n",
       "30  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   GCA_002730035.1   \n",
       "31  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   GCA_002730035.1   \n",
       "\n",
       "    fastani_reference_radius  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        NaN   \n",
       "28                      95.0   \n",
       "29                      95.0   \n",
       "30                      95.0   \n",
       "31                      95.0   \n",
       "\n",
       "                                     fastani_taxonomy  fastani_ani  \\\n",
       "0                                                 NaN          NaN   \n",
       "1                                                 NaN          NaN   \n",
       "2                                                 NaN          NaN   \n",
       "3                                                 NaN          NaN   \n",
       "28  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...         99.8   \n",
       "29  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...         99.8   \n",
       "30  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...         99.8   \n",
       "31  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...         99.8   \n",
       "\n",
       "    fastani_af closest_placement_reference  closest_placement_radius  \\\n",
       "0          NaN                         NaN                       NaN   \n",
       "1          NaN                         NaN                       NaN   \n",
       "2          NaN                         NaN                       NaN   \n",
       "3          NaN                         NaN                       NaN   \n",
       "28       0.964             GCA_002730035.1                      95.0   \n",
       "29       0.964             GCA_002730035.1                      95.0   \n",
       "30       0.964             GCA_002730035.1                      95.0   \n",
       "31       0.964             GCA_002730035.1                      95.0   \n",
       "\n",
       "                           closest_placement_taxonomy  closest_placement_ani  \\\n",
       "0                                                 NaN                    NaN   \n",
       "1                                                 NaN                    NaN   \n",
       "2                                                 NaN                    NaN   \n",
       "3                                                 NaN                    NaN   \n",
       "28  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...                   99.8   \n",
       "29  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...                   99.8   \n",
       "30  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...                   99.8   \n",
       "31  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...                   99.8   \n",
       "\n",
       "    closest_placement_af                                   pplacer_taxonomy  \\\n",
       "0                    NaN  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   \n",
       "1                    NaN  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   \n",
       "2                    NaN  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   \n",
       "3                    NaN  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   \n",
       "28                 0.964  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   \n",
       "29                 0.964  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   \n",
       "30                 0.964  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   \n",
       "31                 0.964  d__Bacteria;p__Pseudomonadota;c__Gammaproteoba...   \n",
       "\n",
       "                                classification_method  \\\n",
       "0   taxonomic classification fully defined by topo...   \n",
       "1   taxonomic classification fully defined by topo...   \n",
       "2   taxonomic classification fully defined by topo...   \n",
       "3   taxonomic classification fully defined by topo...   \n",
       "28  taxonomic classification defined by topology a...   \n",
       "29  taxonomic classification defined by topology a...   \n",
       "30  taxonomic classification defined by topology a...   \n",
       "31  taxonomic classification defined by topology a...   \n",
       "\n",
       "                                                 note  \\\n",
       "0   classification based on placement in class-lev...   \n",
       "1   classification based on placement in class-lev...   \n",
       "2   classification based on placement in class-lev...   \n",
       "3   classification based on placement in class-lev...   \n",
       "28  topological placement and ANI have congruent s...   \n",
       "29  topological placement and ANI have congruent s...   \n",
       "30  topological placement and ANI have congruent s...   \n",
       "31  topological placement and ANI have congruent s...   \n",
       "\n",
       "   other_related_references(genome_id,species_name,radius,ANI,AF)  \\\n",
       "0                                                 NaN               \n",
       "1                                                 NaN               \n",
       "2                                                 NaN               \n",
       "3                                                 NaN               \n",
       "28  GCA_003451715.1, s__UBA9145 sp003451715, 95.0,...               \n",
       "29  GCA_003451715.1, s__UBA9145 sp003451715, 95.0,...               \n",
       "30  GCA_003451715.1, s__UBA9145 sp003451715, 95.0,...               \n",
       "31  GCA_003451715.1, s__UBA9145 sp003451715, 95.0,...               \n",
       "\n",
       "    msa_percent  translation_table  red_value warnings_x    domain  \\\n",
       "0         48.26               11.0    0.75999        NaN  Bacteria   \n",
       "1         48.26               11.0    0.75999        NaN  Bacteria   \n",
       "2         48.26               11.0    0.75999        NaN  Bacteria   \n",
       "3         48.26               11.0    0.75999        NaN  Bacteria   \n",
       "28        73.72               11.0        NaN        NaN  Bacteria   \n",
       "29        73.72               11.0        NaN        NaN  Bacteria   \n",
       "30        73.72               11.0        NaN        NaN  Bacteria   \n",
       "31        73.72               11.0        NaN        NaN  Bacteria   \n",
       "\n",
       "             phyla                class            order               family  \\\n",
       "0   Pseudomonadota  Gammaproteobacteria  Pseudomonadales             HTCC2089   \n",
       "1   Pseudomonadota  Gammaproteobacteria  Pseudomonadales             HTCC2089   \n",
       "2   Pseudomonadota  Gammaproteobacteria  Pseudomonadales             HTCC2089   \n",
       "3   Pseudomonadota  Gammaproteobacteria  Pseudomonadales             HTCC2089   \n",
       "28  Pseudomonadota  Gammaproteobacteria  Pseudomonadales  Pseudohongiellaceae   \n",
       "29  Pseudomonadota  Gammaproteobacteria  Pseudomonadales  Pseudohongiellaceae   \n",
       "30  Pseudomonadota  Gammaproteobacteria  Pseudomonadales  Pseudohongiellaceae   \n",
       "31  Pseudomonadota  Gammaproteobacteria  Pseudomonadales  Pseudohongiellaceae   \n",
       "\n",
       "      genus              species            contig_id  contig_length provirus  \\\n",
       "0       NaN                  NaN  vMAG_32|N=2|L=21616          21816       No   \n",
       "1       NaN                  NaN  vMAG_32|N=3|L=73423          73823       No   \n",
       "2       NaN                  NaN  vMAG_32|N=2|L=59685          59885       No   \n",
       "3       NaN                  NaN  vMAG_32|N=2|L=25053          25253       No   \n",
       "28  UBA9145  UBA9145 sp002730035  vMAG_11|N=2|L=11750          11950       No   \n",
       "29  UBA9145  UBA9145 sp002730035  vMAG_11|N=2|L=53693          53893       No   \n",
       "30  UBA9145  UBA9145 sp002730035  vMAG_11|N=2|L=48736          48936       No   \n",
       "31  UBA9145  UBA9145 sp002730035  vMAG_11|N=2|L=12392          12592       No   \n",
       "\n",
       "    proviral_length  gene_count  viral_genes  host_genes  checkv_quality  \\\n",
       "0               NaN          37            7           2     Low-quality   \n",
       "1               NaN          49            6           0     Low-quality   \n",
       "2               NaN          32           22           0     Low-quality   \n",
       "3               NaN          32            2           1     Low-quality   \n",
       "28              NaN          14            0           1  Not-determined   \n",
       "29              NaN          54            7           0    High-quality   \n",
       "30              NaN          35            3           0  Medium-quality   \n",
       "31              NaN          19            8           0     Low-quality   \n",
       "\n",
       "     miuvig_quality  completeness            completeness_method  \\\n",
       "0   Genome-fragment          5.53  AAI-based (medium-confidence)   \n",
       "1   Genome-fragment         21.85  AAI-based (medium-confidence)   \n",
       "2   Genome-fragment         32.90    AAI-based (high-confidence)   \n",
       "3   Genome-fragment         40.09  AAI-based (medium-confidence)   \n",
       "28  Genome-fragment           NaN                            NaN   \n",
       "29     High-quality        100.00  AAI-based (medium-confidence)   \n",
       "30  Genome-fragment         52.93    AAI-based (high-confidence)   \n",
       "31  Genome-fragment          6.70  AAI-based (medium-confidence)   \n",
       "\n",
       "    contamination  kmer_freq               warnings_y  N      L  \n",
       "0             0.0       1.00                      NaN  2  21616  \n",
       "1             0.0       1.00                      NaN  3  73423  \n",
       "2             0.0       1.00                      NaN  2  59685  \n",
       "3             0.0       1.00                      NaN  2  25053  \n",
       "28            0.0       1.00  no viral genes detected  2  11750  \n",
       "29            0.0       1.01                      NaN  2  53693  \n",
       "30            0.0       1.00                      NaN  2  48736  \n",
       "31            0.0       1.00                      NaN  2  12392  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vMAG_abund"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ebe0046-d6d8-4c2c-968e-f03916417d50",
   "metadata": {},
   "source": [
    "## Criteria: Quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "aa693328-e085-4594-9b29-85f45a579847",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "114"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Virus length\n",
    "long_length = df[df['virus_length'] >= 15000]\n",
    "len(long_length)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b634b701-5e4e-43dc-b603-e555dfcc51e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sort by number of identified viral genes\n",
    "viral_genes = long_length.sort_values(by='viral_genes',ascending=False)\n",
    "viral_genes = viral_genes.dropna(subset='order')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "93585593-7269-480f-a26e-dfbe5194f29a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<AxesSubplot:xlabel='viral_genes', ylabel='Count'>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX8AAAEHCAYAAABGNUbLAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAASIUlEQVR4nO3de5BkZX3G8e8DK0FEEWVE5OJiQogGFXW8AMYEUAsvhZcQkXjBiK4VLxE1GiirYlKppJLSGIxJ1A0iXig04p0YkSBKGQk4u6AuAmoUZQXcUZLgJRFWfvmjz4Zx3NnpHab79Mz7/VR1dZ/TPf0+TC1Pn3m7+z2pKiRJbdml7wCSpPGz/CWpQZa/JDXI8pekBln+ktSgNX0HGMY+++xTa9eu7TuGJK0oGzZs+H5VTW3vvhVR/mvXrmVmZqbvGJK0oiT59kL3Oe0jSQ2y/CWpQZa/JDXI8pekBln+ktQgy1+SGjSy8k9yVpItSTZt574/TFJJ9hnV+JKkhY3yyP9s4Lj5O5McCDwB+M4Ix5Yk7cDIyr+qLgFu3s5dfwO8DvBEApLUk7HO+Sc5HvhuVX1piMeuSzKTZGZ2dnbJY+5/4EEkWZbL/gcetOQckjRJxra8Q5I9gNcDTxzm8VW1HlgPMD09veS/Em7YfD0nvuMLS/3xn/OBlxy5LM8jSX0b55H/LwMHA19Kch1wALAxyX3HmEGSxBiP/KvqK8B9tm13LwDTVfX9cWWQJA2M8qOe5wKXAocm2ZzklFGNJUnaOSM78q+qkxa5f+2oxpYk7Zjf8JWkBln+ktQgy1+SGmT5S1KDLH9JapDlL0kNsvwlqUGWvyQ1yPKXpAZZ/j1xqWlJfRrbwm76eS41LalPHvlLUoMsf0lqkOUvSQ2y/CWpQZa/JDXI8pekBln+ktQgy1+SGmT5S1KDRlb+Sc5KsiXJpjn73pjkmiRfTvKRJPcc1fiSpIWN8sj/bOC4efsuBA6rqocAXwNOH+H4kqQFjKz8q+oS4OZ5+z5dVVu7zX8HDhjV+JKkhfU55/9C4F8WujPJuiQzSWZmZ2fHGEuSVr9eyj/J64GtwDkLPaaq1lfVdFVNT01NjS+cJDVg7Es6JzkZeCpwbFXVuMeXJI25/JMcB/wR8JtV9ZNxji1JusMoP+p5LnApcGiSzUlOAf4OuDtwYZIrk7x9VONLkhY2siP/qjppO7vfOarxJEnD8xu+ktQgy1+SGmT5S1KDLH9JapDlL0kNsvwlqUGWvyQ1yPKXpAZZ/pLUoLEv7Lai7bKGJH2nkKQ7zfLfGbdv5cR3fGFZnuoDLzlyWZ5HkpbCaR9JapDlL0kNsvwlqUGWvyQ1yPKXpAZZ/pLUIMtfkhpk+UtSgyx/SWqQ5S9JDRpZ+Sc5K8mWJJvm7LtXkguTfL273ntU40uSFjbKI/+zgePm7TsNuKiqDgEu6rYlSWM2svKvqkuAm+ftfhrw7u72u4Gnj2p8SdLCxj3nv29V3QjQXd9noQcmWZdkJsnM7Ozs2AJKUgsm9g3fqlpfVdNVNT01NdV3HElaVcZd/t9Lsh9Ad71lzONLkhh/+X8cOLm7fTLwsTGPL0litB/1PBe4FDg0yeYkpwB/CTwhydeBJ3TbkqQxG9lpHKvqpAXuOnZUY0qShjOxb/hKkkbH8pekBln+ktQgy1+SGmT5S1KDLH9JapDlL0kNsvwlqUGWvyQ1yPKXpAZZ/pLUIMtfkhpk+UtSgyx/SWqQ5S9JDbL8JalBlr8kNcjyl6QGWf6S1CDLX5Ia1Ev5J3lVkquSbEpybpLd+8ghSa0aqvyTHDXMviGfa3/gD4DpqjoM2BV49lKeS5K0NMMe+b91yH3DWgPcNckaYA/ghjvxXJKknbRmR3cmOQI4EphK8uo5d92DwRH7Tquq7yZ5E/Ad4H+AT1fVp5fyXJKkpVnsyH83YE8GLxJ3n3O5BThhKQMm2Rt4GnAwcD/gbkmeu53HrUsyk2RmdnZ2KUNJkhawwyP/qvoc8LkkZ1fVt5dpzMcD36qqWYAkH2bw18X75o29HlgPMD09Xcs0tiSJRcp/jl9Ksh5YO/dnquqYJYz5HeAxSfZgMO1zLDCzhOeRJC3RsOX/QeDtwJnAz+7MgFV1WZLzgI3AVuAKuiN8SdJ4DFv+W6vqbcs1aFW9AXjDcj2fJGnnDPtRz08keWmS/ZLca9tlpMkkSSMz7JH/yd31a+fsK+AByxtHkjQOQ5V/VR086iCSpPEZqvyTPH97+6vqPcsbR5I0DsNO+zxyzu3dGXw8cyNg+UvSCjTstM8r5m4n2Qt470gSSZJGbqlLOv8EOGQ5g0iSxmfYOf9PMPh0DwwWdHsg8E+jCiVJGq1h5/zfNOf2VuDbVbV5BHkkSWMw1LRPt8DbNQxW9NwbuHWUoSRJozXsmbyeBVwO/A7wLOCyJEta0lmS1L9hp31eDzyyqrYAJJkC/hU4b1TBJEmjM+ynfXbZVvydH+zEz0qSJsywR/6fSnIBcG63fSLwydFEkiSN2mLn8P0VYN+qem2SZwKPBQJcCpwzhnySpBFYbOrmDOCHAFX14ap6dVW9isFR/xmjjSZJGpXFyn9tVX15/s6qmmFwSkdJ0gq0WPnvvoP77rqcQSRJ47NY+X8xyYvn70xyCrBhNJEkSaO22Kd9TgU+kuQ53FH208BuwDNGmEuSNEI7LP+q+h5wZJKjgcO63f9cVZ8ZeTJJ0sgMu57/xcDFyzVoknsCZzJ4QSnghVV16XI9vyRpx4b9ktdyewvwqao6IcluwB495ZCkJo29/JPcA3gc8AKAqroVVwmVpLHqY32eBwCzwLuSXJHkzCR3m/+gJOuSzCSZmZ2dHX9KSVrF+ij/NcDDgbdV1cOAHwOnzX9QVa2vqumqmp6amhp3Rkla1foo/83A5qq6rNs+j8GLgSRpTMZe/lV1E3B9kkO7XccCXx13DklqWV+f9nkFcE73SZ9vAr/XUw5JalIv5V9VVzL4prAkqQeejUuSGmT5S1KDLH9JapDlL0kNsvwlqUGWvyQ1yPKXpAZZ/pLUIMtfkhpk+UtSgyx/SWqQ5S9JDbL8JalBlr8kNcjyl6QGWf6S1CDLX5IaZPlLUoMsf0lqkOUvSQ3qrfyT7JrkiiTn95VBklrV55H/K4GrexxfkprVS/knOQB4CnBmH+NLUuv6OvI/A3gdcHtP40tS08Ze/kmeCmypqg2LPG5dkpkkM7Ozs2NKJ0lt6OPI/yjg+CTXAe8HjknyvvkPqqr1VTVdVdNTU1PjzihJq9rYy7+qTq+qA6pqLfBs4DNV9dxx55Cklvk5f0lq0Jo+B6+qzwKf7TODJLXII39JapDlL0kNsvwlqUGWvyQ1yPKXpAZZ/pLUIMtfkhpk+UtSgyx/SWqQ5S9JDbL8JalBlr8kNcjyl6QGWf6S1CDLX5IaZPlLUoMsf0lqkOUvSQ2y/CWpQZa/JDXI8pekBo29/JMcmOTiJFcnuSrJK8edQZJat6aHMbcCr6mqjUnuDmxIcmFVfbWHLJLUpLEf+VfVjVW1sbv9Q+BqYP9x55CklvU6559kLfAw4LLt3LcuyUySmdnZ2bFna9X+Bx5EkmW5rNlt92V7rv0PPKjvX812Lefva1L/G7U69THtA0CSPYEPAadW1S3z76+q9cB6gOnp6RpzvGbdsPl6TnzHF5bluT7wkiOX9bkm0XL/vqRx6eXIP8ldGBT/OVX14T4ySFLL+vi0T4B3AldX1ZvHPb4kqZ8j/6OA5wHHJLmyuzy5hxyS1Kyxz/lX1eeBjHtcSdId/IavJDXI8pekBln+ktQgy1+SGmT5S1KDLH9JapDlL0kNsvwlqUGWvyQ1yPJfDXZZs2zLCrdgOZdhbsGkLls9qbkmPds2vS3prGV0+1aXFd4JLsO8cyb19zWpuWCys23jkb8kNcjyl6QGWf6S1CDLX5IaZPlLUoMsf0lqkOUvSQ2y/CWpQZa/JDXI8pekBvVS/kmOS3Jtkm8kOa2PDJLUsrGXf5Jdgb8HngQ8CDgpyYPGnUOSWtbHkf+jgG9U1Ter6lbg/cDTesghSc1KVY13wOQE4LiqelG3/Tzg0VX18nmPWwes6zYPBa5d4pD7AN9f4s/2xcyjt9LygpnHZaVl3lHe+1fV1Pbu6GNJ5+0tgv4Lr0BVtR5Yf6cHS2aqavrOPs84mXn0VlpeMPO4rLTMS83bx7TPZuDAOdsHADf0kEOSmtVH+X8ROCTJwUl2A54NfLyHHJLUrLFP+1TV1iQvBy4AdgXOqqqrRjjknZ466oGZR2+l5QUzj8tKy7ykvGN/w1eS1D+/4StJDbL8JalBq7r8V8IyEknOSrIlyaY5++6V5MIkX++u9+4z41xJDkxycZKrk1yV5JXd/knOvHuSy5N8qcv8p93+ic0Mg2/DJ7kiyfnd9qTnvS7JV5JcmWSm2zfpme+Z5Lwk13T/po+Y5MxJDu1+v9sutyQ5dSmZV235r6BlJM4Gjpu37zTgoqo6BLio254UW4HXVNUDgccAL+t+r5Oc+afAMVX1UOBw4Lgkj2GyMwO8Erh6zvak5wU4uqoOn/O580nP/BbgU1X1a8BDGfy+JzZzVV3b/X4PBx4B/AT4CEvJXFWr8gIcAVwwZ/t04PS+cy2QdS2wac72tcB+3e39gGv7zriD7B8DnrBSMgN7ABuBR09yZgbff7kIOAY4fyX8uwCuA/aZt29iMwP3AL5F98GXlZB5Xs4nAv+21Myr9sgf2B+4fs725m7fSrBvVd0I0F3fp+c825VkLfAw4DImPHM3hXIlsAW4sKomPfMZwOuA2+fsm+S8MPim/qeTbOiWZ4HJzvwAYBZ4Vze9dmaSuzHZmed6NnBud3unM6/m8h9qGQktTZI9gQ8Bp1bVLX3nWUxV/awGfyofADwqyWE9R1pQkqcCW6pqQ99ZdtJRVfVwBlOtL0vyuL4DLWIN8HDgbVX1MODHTNAUz450X5A9HvjgUp9jNZf/Sl5G4ntJ9gPorrf0nOfnJLkLg+I/p6o+3O2e6MzbVNV/AZ9l8D7LpGY+Cjg+yXUMVr09Jsn7mNy8AFTVDd31Fgbz0I9isjNvBjZ3fwUCnMfgxWCSM2/zJGBjVX2v297pzKu5/FfyMhIfB07ubp/MYF59IiQJ8E7g6qp685y7JjnzVJJ7drfvCjweuIYJzVxVp1fVAVW1lsG/289U1XOZ0LwASe6W5O7bbjOYj97EBGeuqpuA65Mc2u06FvgqE5x5jpO4Y8oHlpK57zctRvyGyJOBrwH/Aby+7zwLZDwXuBG4jcGRyCnAvRm82ff17vpefeeck/exDKbPvgxc2V2ePOGZHwJc0WXeBPxxt39iM8/J/lvc8YbvxOZlMH/+pe5y1bb/3yY5c5fvcGCm+7fxUWDvFZB5D+AHwF5z9u10Zpd3kKQGreZpH0nSAix/SWqQ5S9JDbL8JalBlr8kNcjyl6QGWf5a9ZLcL8l5S/i5H40ijzQJ/Jy/mpVkTVVt3cH9P6qqPceZSRoXj/y1qiT5qyQvnbP9J0les+1kOUlekOSDST7BYAXKPZNclGRjdyKSpw05zi5J/qE7Ocz5ST6Z5ITuvkck+Vy3uuUFc9Zc+WyX7/IkX0vyG93+XZO8MckXk3w5yUu6/fsluaQ7acembY+XloPlr9Xm/cCJc7afxWCdp7mOAE6uqmOA/wWeUYPVKI8G/rpbv2gxz2RwHoYHAy/qnnPbondvBU6oqkcAZwF/Pufn1lTVo4BTgTd0+04B/ruqHgk8EnhxkoOB32VwTorDGZxo5MohcklDWdN3AGk5VdUVSe6T5H7AFPCfwHfmPezCqrq5ux3gL7rlh29ncM6HfYGbFhnqscAHq+p24KYkF3f7DwUOAy7sXkN2ZbB20zbbVkHdwODFAwaLoD1k218OwF7AIQxetM7qXlA+WlVXLpJJGprlr9XoPOAE4L4M/hKY78dzbj+HwYvEI6rqtm4Z5d2HGGOhvw4CXFVVRyxw/0+7659xx/9/AV5RVRf8wpMNXpSeArw3yRur6j1DZJMW5bSPVqP3M1gK+QQGLwQ7sheDE6fcluRo4P5DjvF54Le7uf99Gay+CYPT6U0l+f9poCS/vshzXQD8fneET5Jf7ZZIvn+X7R8ZLKP98CGzSYvyyF+rTlVd1a0t/92qurE73eRCzgE+kWSGwZz6NUMO8yEG679vYrBs+GUM5u1v7aZv/jbJXgz+HzuDwTLHCzmTwRTQxu79hlng6QxeUF6b5DbgR8Dzh8wmLcqPekpLlGTPqvpRknsDlzM4jeFi7xVIE8Ejf2npzu/OELYb8GcWv1YSj/ylHUjyYOC983b/tKoe3UceablY/pLUID/tI0kNsvwlqUGWvyQ1yPKXpAb9H1sYd4kAZ8y1AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# histogram of viral gene numbers\n",
    "sns.histplot(viral_genes['viral_genes'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "552ef9be-5620-4464-8075-af3b95643f86",
   "metadata": {},
   "source": [
    "## Connection Strength"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "812ab471-abbf-4bae-8dc5-f7cee179488b",
   "metadata": {},
   "source": [
    "Proposed criteria: Adjusted inter connective linkage density >= 0.05 reads/kbp^2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "109f118c-0c03-472d-9643-1676da406c08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "87 associations are >= 0.05, which is about 26 %.\n"
     ]
    }
   ],
   "source": [
    "inter = df[df['adjusted_inter_linkage_density'] >= 0.05]\n",
    "print(len(inter), 'associations are >= 0.05, which is about', round(len(inter)/len(df)*100), '%.')"
   ]
  },
  {
   "cell_type": "raw",
   "id": "31c56bdd-0db4-42f4-8f36-1ccbd9e2fac0",
   "metadata": {},
   "source": [
    "Proposed criteria: inter vs intra ratio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "35513b99-a829-4ac3-a264-0fb3c597713e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "88 associations are >= 1, which is about 26 %.\n"
     ]
    }
   ],
   "source": [
    "ratio = df[df['adjusted_inter_vs_intra_ratio'] >= 2.5]\n",
    "print(len(ratio), 'associations are >= 1, which is about', round(len(ratio)/len(df)*100), '%.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "80d5d0dd-a756-47ab-be53-bda88bbcb3fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>virus_name</th>\n",
       "      <th>virus_length</th>\n",
       "      <th>virus_read_count</th>\n",
       "      <th>virus_read_depth</th>\n",
       "      <th>virus_read_depth_in_host</th>\n",
       "      <th>host_name</th>\n",
       "      <th>host_length</th>\n",
       "      <th>host_read_count</th>\n",
       "      <th>host_read_depth</th>\n",
       "      <th>intra_read_count</th>\n",
       "      <th>...</th>\n",
       "      <th>translation_table</th>\n",
       "      <th>red_value</th>\n",
       "      <th>warnings</th>\n",
       "      <th>domain</th>\n",
       "      <th>phyla</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>k141_3298423</td>\n",
       "      <td>5405</td>\n",
       "      <td>231</td>\n",
       "      <td>42.738205</td>\n",
       "      <td>15.114731</td>\n",
       "      <td>bin_110</td>\n",
       "      <td>155303</td>\n",
       "      <td>4643</td>\n",
       "      <td>29.896396</td>\n",
       "      <td>142</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No bacterial or archaeal marker</td>\n",
       "      <td>Unclassified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>k141_2834672</td>\n",
       "      <td>16753</td>\n",
       "      <td>366</td>\n",
       "      <td>21.846833</td>\n",
       "      <td>8.516562</td>\n",
       "      <td>bin_135</td>\n",
       "      <td>110072</td>\n",
       "      <td>2484</td>\n",
       "      <td>22.567047</td>\n",
       "      <td>106</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No bacterial or archaeal marker</td>\n",
       "      <td>Unclassified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       virus_name  virus_length  virus_read_count  virus_read_depth  \\\n",
       "209  k141_3298423          5405               231         42.738205   \n",
       "286  k141_2834672         16753               366         21.846833   \n",
       "\n",
       "     virus_read_depth_in_host host_name  host_length  host_read_count  \\\n",
       "209                 15.114731   bin_110       155303             4643   \n",
       "286                  8.516562   bin_135       110072             2484   \n",
       "\n",
       "     host_read_depth  intra_read_count  ...  translation_table  red_value  \\\n",
       "209        29.896396               142  ...                NaN        NaN   \n",
       "286        22.567047               106  ...                NaN        NaN   \n",
       "\n",
       "                            warnings        domain  phyla  class  order  \\\n",
       "209  No bacterial or archaeal marker  Unclassified    NaN    NaN    NaN   \n",
       "286  No bacterial or archaeal marker  Unclassified    NaN    NaN    NaN   \n",
       "\n",
       "    family genus  species  \n",
       "209    NaN   NaN      NaN  \n",
       "286    NaN   NaN      NaN  \n",
       "\n",
       "[2 rows x 46 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "both = df[(df['adjusted_inter_linkage_density'] >= 0.05) & (df['adjusted_inter_vs_intra_ratio'] >= 2.5)]\n",
    "both"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d279748e-cde2-4080-8492-19177b145e06",
   "metadata": {},
   "source": [
    "There are only 2 associations that have adjusted_inter_linkage_density >= 0.05 and adjusted_inter_vs_intra_ratio >= 2.5. So I recommend picking one of the criteria, not both."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd4bbe80-4368-4704-ab7a-46857d00dcc5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
