{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "199f4a08-e6ec-4f34-a1d6-4a00f49689a7",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    "\n",
    "Notebooks are listed in the order they should be run in.\n",
    "\n",
    "***00_functions.py*** : a list of all functions (with notes and comments) used for this project compiled into one python document for easy copying and pasting into jupyter notebooks\n",
    "\n",
    "## scripts\n",
    "***01_create_proximeta_associations_table*** : creates a master table of all proximeta data \n",
    "\n",
    "***02_create_sag_master_table*** : creates a master table of all sag data\n",
    "\n",
    "## preliminary_analysis\n",
    "\n",
    "***01_proximeta_summary*** : summarizes the master proximeta virus table\n",
    "+ Virus type breakdown and plot\n",
    "+ Strength of host-virus associations\n",
    "    + plots: distribution of adjusted_inter_vs_intra_ratio\n",
    "    + distribution of adjusted_inter_connective_linkage_density\n",
    "+ Viral copies per cell\n",
    "    + plots: jitter and boxplot\n",
    "+ number of associated vmags and the total number of vmags\n",
    "\n",
    "***02_sag_summary*** : plot of the number of associated vsags and the total number of vsags\n",
    "+ boxplot\n",
    "\n",
    "## gtdb-tk\n",
    "\n",
    "***04_MAG_gtdb*** : phylogenomic analysis of all mags (with and without viruses associated to them)\n",
    "+ Input merged mag gtdb table from 03_create_merged_mag_gtdb_table notebook\n",
    "+ Create lists of ten most abundant taxonomic groups for each level of classification\n",
    "+ How many MAGs were classified in total?\n",
    "+ How many MAGs were classified to at least each classification level?\n",
    "    + plots: Count and Percent\n",
    "+ What are the most common classifications?\n",
    "    + plots: count of top 10 most abundant phyla, classes, and orders\n",
    "+ Plots: stacked barplots of top order abundances \n",
    "    + count and percent\n",
    " \n",
    "<u><b> Notes </b></u> : fix color map code to match between all gtdb analysis notebooks\n",
    "\n",
    "***06_proximeta_assoc_gtdbtk*** : phylogenomic analysis of mags associated with viruses\n",
    "+ Input proximeta associations gtdb table from 05_create_proximeta_assoc_gtdb_table notebook\n",
    "+ Create lists of ten most abundant taxonomic groups for each level of classification\n",
    "+ How many associations were classified in total?\n",
    "+ How many associations were classified to at least each classification level?\n",
    "    + plots: Count and Percent\n",
    "+ What are the most common classifications?\n",
    "    + plots: count of top 10 most abundant phyla, classes, and orders\n",
    "+ Plots: stacked barplots of top order abundances \n",
    "    + count and percent\n",
    "    \n",
    "<u><b> Notes </b></u> : fix color map code to match between all gtdb analysis notebooks\n",
    "    \n",
    "***07_sag_assoc_gtdbtk*** : phylogenomic analysis of sags associated with viruses\n",
    "+ Input sag associations gtdb file from sag_data folder\n",
    "+ Create lists of ten most abundant taxonomic groups for each level of classification\n",
    "+ How many associations were classified in total?\n",
    "+ How many associations were classified to at least each classification level?\n",
    "    + plots: Count and Percent\n",
    "+ What are the most common classifications?\n",
    "    + plots: count of top 10 most abundant phyla, classes, and orders\n",
    "+ Plots: stacked barplots of top order abundances \n",
    "    + count and percent\n",
    "    \n",
    "<u><b> Notes </b></u> : fix color map code to match between all gtdb analysis notebooks\n",
    "\n",
    "***09_sag_vs_mag_host_tax*** : compare host phylogeny between associated sags and mags\n",
    "+\n",
    "\n",
    "<u><b> Notes </b></u> : fix color map code to match between all gtdb analysis notebooks\n",
    "\n",
    "\n",
    "## genome_annotation\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4eeb8b74-202c-4bcd-a801-e01627f04ee0",
   "metadata": {},
   "source": [
    "To do:\n",
    "+ fix color maps\n",
    "+ figure interpretations\n",
    "\n",
    "\n",
    "1 large table:\n",
    "+ sample name\n",
    "+ sample depth\n",
    "+ virus type: sag, mag, or contig\n",
    "+ associated host or not\n",
    "+ gtdbtk\n",
    "+ summary columns\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
