{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bf67237f-9372-44a9-b8db-6f9b7c2db16c",
   "metadata": {},
   "source": [
    "# 01_proximeta_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0d8fed73-4694-419b-a933-11c1b19133c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load packages\n",
    "import pandas as pd\n",
    "import os\n",
    "import os.path as op\n",
    "import sys\n",
    "import csv\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import glob\n",
    "import seaborn as sns\n",
    "from collections import Counter\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# file paths\n",
    "path = '/Users/melissaherring/Google Drive/My Drive/MH_project/'\n",
    "#path = '/Users/juliabrown/Google Drive/My Drive/projects/OMZvir_round2/MH_project/'\n",
    "\n",
    "# load proximeta master table\n",
    "proximeta = pd.read_csv(op.join(path, 'tables/proximeta_master.csv'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "235251fe-4cb6-4b35-b4b1-4b5f1163858e",
   "metadata": {},
   "source": [
    "## Virus type breakdown:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "dd7ad017-267b-42ff-8090-621822251d7c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>virus_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>contig</td>\n",
       "      <td>295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>unassociated vmag</td>\n",
       "      <td>195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uninfected mag</td>\n",
       "      <td>535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>vmag</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          virus_type  count\n",
       "0             contig    295\n",
       "1  unassociated vmag    195\n",
       "2     uninfected mag    535\n",
       "3               vmag     41"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proximeta.groupby('virus_type').size().reset_index(name='count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77988548-c8ec-4598-9516-bfbd181891ec",
   "metadata": {},
   "source": [
    "In this dataset, there are 295 viral contigs and 41 vmags that are associated with hosts. There are 195 vmags that are not associated with hosts. Lastly, there are 535 mags that are not associated with viruses."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cf213878-566b-41dd-855e-9225e0004897",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>virus_name</th>\n",
       "      <th>virus_length</th>\n",
       "      <th>virus_read_count</th>\n",
       "      <th>virus_read_depth</th>\n",
       "      <th>virus_read_depth_in_host</th>\n",
       "      <th>host_name</th>\n",
       "      <th>host_length</th>\n",
       "      <th>host_read_count</th>\n",
       "      <th>host_read_depth</th>\n",
       "      <th>intra_read_count</th>\n",
       "      <th>...</th>\n",
       "      <th>host_genes</th>\n",
       "      <th>checkv_quality</th>\n",
       "      <th>miuvig_quality</th>\n",
       "      <th>completeness</th>\n",
       "      <th>completeness_method</th>\n",
       "      <th>contamination</th>\n",
       "      <th>kmer_freq</th>\n",
       "      <th>warnings_y</th>\n",
       "      <th>N</th>\n",
       "      <th>L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>871</th>\n",
       "      <td>vMAG_7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>66.90</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>36972.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>872</th>\n",
       "      <td>vMAG_12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>70.44</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>30847.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>873</th>\n",
       "      <td>vMAG_36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>3.68</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13927.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>vMAG_8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>54.88</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>30359.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>vMAG_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>80.00</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>33140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1061</th>\n",
       "      <td>vMAG_35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>28.14</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15980.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1062</th>\n",
       "      <td>vMAG_45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>36.45</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>14520.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1063</th>\n",
       "      <td>vMAG_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>57.21</td>\n",
       "      <td>AAI-based (high-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33980.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1064</th>\n",
       "      <td>vMAG_27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Medium-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>75.07</td>\n",
       "      <td>AAI-based (medium-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>33970.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065</th>\n",
       "      <td>vMAG_40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Low-quality</td>\n",
       "      <td>Genome-fragment</td>\n",
       "      <td>39.75</td>\n",
       "      <td>AAI-based (high-confidence)</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>22570.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>195 rows Ã— 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     virus_name  virus_length  virus_read_count  virus_read_depth  \\\n",
       "871      vMAG_7           NaN               NaN               NaN   \n",
       "872     vMAG_12           NaN               NaN               NaN   \n",
       "873     vMAG_36           NaN               NaN               NaN   \n",
       "874      vMAG_8           NaN               NaN               NaN   \n",
       "875      vMAG_2           NaN               NaN               NaN   \n",
       "...         ...           ...               ...               ...   \n",
       "1061    vMAG_35           NaN               NaN               NaN   \n",
       "1062    vMAG_45           NaN               NaN               NaN   \n",
       "1063     vMAG_3           NaN               NaN               NaN   \n",
       "1064    vMAG_27           NaN               NaN               NaN   \n",
       "1065    vMAG_40           NaN               NaN               NaN   \n",
       "\n",
       "      virus_read_depth_in_host host_name  host_length  host_read_count  \\\n",
       "871                        NaN       NaN          NaN              NaN   \n",
       "872                        NaN       NaN          NaN              NaN   \n",
       "873                        NaN       NaN          NaN              NaN   \n",
       "874                        NaN       NaN          NaN              NaN   \n",
       "875                        NaN       NaN          NaN              NaN   \n",
       "...                        ...       ...          ...              ...   \n",
       "1061                       NaN       NaN          NaN              NaN   \n",
       "1062                       NaN       NaN          NaN              NaN   \n",
       "1063                       NaN       NaN          NaN              NaN   \n",
       "1064                       NaN       NaN          NaN              NaN   \n",
       "1065                       NaN       NaN          NaN              NaN   \n",
       "\n",
       "      host_read_depth  intra_read_count  ...  host_genes  checkv_quality  \\\n",
       "871               NaN               NaN  ...         0.0  Medium-quality   \n",
       "872               NaN               NaN  ...         4.0  Medium-quality   \n",
       "873               NaN               NaN  ...         0.0     Low-quality   \n",
       "874               NaN               NaN  ...         0.0  Medium-quality   \n",
       "875               NaN               NaN  ...         0.0  Medium-quality   \n",
       "...               ...               ...  ...         ...             ...   \n",
       "1061              NaN               NaN  ...         1.0     Low-quality   \n",
       "1062              NaN               NaN  ...         0.0     Low-quality   \n",
       "1063              NaN               NaN  ...         0.0  Medium-quality   \n",
       "1064              NaN               NaN  ...         2.0  Medium-quality   \n",
       "1065              NaN               NaN  ...         1.0     Low-quality   \n",
       "\n",
       "       miuvig_quality  completeness            completeness_method  \\\n",
       "871   Genome-fragment         66.90  AAI-based (medium-confidence)   \n",
       "872   Genome-fragment         70.44  AAI-based (medium-confidence)   \n",
       "873   Genome-fragment          3.68  AAI-based (medium-confidence)   \n",
       "874   Genome-fragment         54.88  AAI-based (medium-confidence)   \n",
       "875   Genome-fragment         80.00  AAI-based (medium-confidence)   \n",
       "...               ...           ...                            ...   \n",
       "1061  Genome-fragment         28.14  AAI-based (medium-confidence)   \n",
       "1062  Genome-fragment         36.45  AAI-based (medium-confidence)   \n",
       "1063  Genome-fragment         57.21    AAI-based (high-confidence)   \n",
       "1064  Genome-fragment         75.07  AAI-based (medium-confidence)   \n",
       "1065  Genome-fragment         39.75    AAI-based (high-confidence)   \n",
       "\n",
       "      contamination  kmer_freq warnings_y    N        L  \n",
       "871             0.0       1.00        NaN  2.0  36972.0  \n",
       "872             0.0       1.00        NaN  2.0  30847.0  \n",
       "873             0.0       1.03        NaN  2.0  13927.0  \n",
       "874             0.0       1.00        NaN  2.0  30359.0  \n",
       "875             0.0       1.00        NaN  3.0  33140.0  \n",
       "...             ...        ...        ...  ...      ...  \n",
       "1061            0.0       1.00        NaN  2.0  15980.0  \n",
       "1062            0.0       1.00        NaN  2.0  14520.0  \n",
       "1063            0.0       1.00        NaN  2.0  33980.0  \n",
       "1064            0.0       1.00        NaN  2.0  33970.0  \n",
       "1065            0.0       1.00        NaN  2.0  22570.0  \n",
       "\n",
       "[195 rows x 61 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proximeta[proximeta['virus_type'] == 'unassociated vmag']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "54dafa9c-d8ae-43f4-80d1-40eaa02584d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_depth</th>\n",
       "      <th>virus_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>80.0</td>\n",
       "      <td>contig</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>80.0</td>\n",
       "      <td>uninfected mag</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>80.0</td>\n",
       "      <td>vmag</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>95.0</td>\n",
       "      <td>contig</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>95.0</td>\n",
       "      <td>uninfected mag</td>\n",
       "      <td>143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>95.0</td>\n",
       "      <td>vmag</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>140.0</td>\n",
       "      <td>contig</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>140.0</td>\n",
       "      <td>uninfected mag</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>140.0</td>\n",
       "      <td>vmag</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>400.0</td>\n",
       "      <td>contig</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>400.0</td>\n",
       "      <td>uninfected mag</td>\n",
       "      <td>152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>400.0</td>\n",
       "      <td>vmag</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_depth      virus_type  count\n",
       "0           80.0          contig     81\n",
       "1           80.0  uninfected mag    129\n",
       "2           80.0            vmag     15\n",
       "3           95.0          contig     77\n",
       "4           95.0  uninfected mag    143\n",
       "5           95.0            vmag     10\n",
       "6          140.0          contig     47\n",
       "7          140.0  uninfected mag    111\n",
       "8          140.0            vmag      9\n",
       "9          400.0          contig     90\n",
       "10         400.0  uninfected mag    152\n",
       "11         400.0            vmag      7"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "virus_count = pd.DataFrame(proximeta.groupby(['sample_depth','virus_type']).size().reset_index(name='count'))\n",
    "virus_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "91e96ed0-f24e-44f2-8c77-338569d6b42a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Text(0, 0.5, 'Count')"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAg0AAAEGCAYAAADmN6bKAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAAhv0lEQVR4nO3deXhV5bn38d8dwiiiBEMEAgSVKSYFJY0FrUqd0INILSgivvG0Hi4HTp1Oe3BocaKlp3Js1VcLRyk4ggVbqa+XVpGhx7ZgUBFIIKQVMRJIFC0gGAi53z/2im7jDixCdnay8/1cF9fOetZ05yHKL88aHnN3AQAAHEpKogsAAAAtA6EBAACEQmgAAAChEBoAAEAohAYAABBKaqILOBLHHXecZ2VlJboMAGhRVq9e/ZG7pye6DrQ8LTo0ZGVlqbCwMNFlAECLYmbvJ7oGtExcngAAAKEQGgAAQCiEBgAAEEqLvqcBANB8rF69untqaupjknLEL6UtVY2kddXV1dcMGzasou5KQgMAoFGkpqY+dvzxxw9OT0//JCUlhYmNWqCamhqrrKzM3rZt22OSxtRdTxIEADSWnPT09J0EhpYrJSXF09PT/6nIaNHX1zdxPQCA5JVCYGj5gr/DmPmA0AAAAELhngYAQFwU5uUPa8zj5RWuWt2Yx4tl48aN7ZYuXdr52muv3SFJK1as6DRnzpxuc+fO/SDe524JCA0AcBCjfrIgLsd9+d7L43JcHJlNmza1X7BgQVptaDjzzDP3nHnmmXsSXVdzweUJAEDSePjhh7sNGDAge+DAgdljx47tV1JS0m748OEDBgwYkD18+PABmzZtaidJ3/ve97Kuvvrq3qeccsqgzMzM3N/+9rddJemOO+7oVVhY2HnQoEHZd999d/cXX3zx6JEjR54kSVu3bk0dMWJE/+zs7METJ07s27Nnz9zy8vJW9cs3oQEAkBQKCws73H///T2WL19esnHjxqJZs2Ztufbaa/tMnDjx45KSkqLLL7/84+uuu6537fbbt29vW1hYuOGFF17YNG3atF6SNH369A/z8vJ2b9iwoWjatGlfeU/B1KlTe5511lm7ioqKii+99NJPysvL2zX195hohAYAQFJ45ZVXulx88cWf9OjRo1qSMjIyDrz99ttHTZ48eYckXXfddTtWr17duXb7MWPGfNqmTRsNGzbs848//rjtoY6/atWqzgUFBTskady4cTu7dOlyIF7fS3NFaAAAJAV3l5mFfuSzQ4cOX2zrfujdwmyT7AgNAICkMGrUqJ2LFy9O27ZtWxtJ2r59e5tTTjnls8cee6yrJM2aNSstLy9v98GOccwxxxzYvXt3m1jr8vPzdz/55JNpkvT888932blzZ8ztklmruoEDANB0muIRya+cLy/v81tvvbX829/+9qCUlBTPycnZ8+ijj24pKCjI+vWvf318t27dqp944onNBztGfn7+3tTUVB84cGD2xIkTPxo2bNje2nUzZszYOm7cuBOys7O7Dh8+fHd6evr+Y489tlVdorCWPNySl5fnhYWFiS4DQBJLxkcuzWy1u+c19nHXrFmzeciQIR819nGbi71791pqaqq3bdtWr7322lFTpkzpu2HDhqJE1xUPa9asOW7IkCFZddsZaQAAIITS0tJ2l1122Yk1NTVq27atz5o1a3Oia2pqhAYAAELIzc2tKi4uTsqRhbC4ERIAAIRCaAAAAKFweQLNQjLebAYAySZuIw1mNsfMKsxsXYx1/2FmbmbHRbXdZmalZrbRzC6IV10AAKBh4jnSMFfSw5KeiG40s96SzpO0JaotW9IESSdL6inpNTMb4O6t6vlXAEgmo36yoFGnxn753ssb/b0PYae+vu+++7rPmTMnPScnZ8/ixYvfO5xz3HPPPd1vvvnmj44++uiasPu8+OKLR8+cOTNj6dKlpYdzrniL20iDu6+QtCPGqgck/VhS9AsiLpE0392r3P09SaWS8uNVGwAAUmTq60MFBkl6/PHH01966aVNhxsYJGnWrFkZu3fvTop7CJv0mzCzMZI+dPc1dVb1khT9l1YWtMU6xmQzKzSzwsrKyjhVCgBoaTZu3Niuf//+J9cu//SnP8245ZZbekpSfn7+wOuuu65Xbm7u4KysrJyXX365sxT5jb526utbbrml5/jx47Py8/MHZmZm5t53333dJWnixIl9ysrK2o8ZM+aku+++u/vOnTtTxo8fn5WTkzN48ODB2U899dSxklRdXa3JkydnDhgwIHvAgAHZ06dP737fffd1r6ioaHvWWWcNOO200wZIkVdQDx06dFB2dvbgCy+88IR//vOfKZK0cOHCLv369Tt52LBhAxcuXHhsrO/xwQcf7Hbuueee+J3vfOekXr165f7sZz9Lv+uuuzIGDx6cPWTIkEHbt29vI0kzZ848LicnZ/DAgQOzL7jgghN37dqVIknr169vP2TIkEE5OTmDb7rppp6dOnU65XD6uMlCg5l1knSHpJ/GWh2jLearKt19trvnuXteenp6Y5YIAEhi1dXVtnbt2uJf/OIXH9xzzz09Y21TWlraYfny5SVvvvlm8f3339+zqqrKnnnmmS3du3ffv3z58pJp06ZV3H777T1Gjhy5c926dcV//vOfN955552ZO3fuTJk5c2b6+++/3379+vVFJSUlRddcc83Hd955Z0XtvitXriwpLy9P/dnPftZjxYoVJUVFRcWnnnrqnnvvvTdjz549NmXKlKzFixeXvvnmmxsrKirqnXWzpKSk46JFi/7x5ptvFv/85z/v1alTp5ri4uKivLy8z2bNmtVNkq688spP1q1bV7xx48aigQMH7n3wwQePk6QpU6b0vv766yvWrVtX3LNnz/2H24dNOdJwoqR+ktaY2WZJmZLeMrPjFRlZ6B21baakrU1YGwAgyY0fP/4TSRoxYsRnZWVl7WJtc/7553/asWNH79GjR3VaWtr+srKyr937t2zZsi4PPPBAj0GDBmWfccYZA6uqqqy0tLTd66+/3uXaa6+tbNs28u99RkbG1+7LW7Zs2VF///vfO+Tn5w8aNGhQ9vz587tt2bKl3TvvvNMhMzOzKjc3tyolJUVXXnnlx/V9HyNGjNjVtWvXmp49e1Z37tz5wPjx4z+VpNzc3D2bN29uL0mrV6/uOGzYsIEDBgzIXrRoUbf169d3kKS333678/e///0dknTNNdfUe476NNkjl+6+VlL32uUgOOS5+0dmtljSM2b234rcCNlf0qqmqg0A0PKlpqZ6Tc2X9xp+/vnnX/nFuHYq7NTUVB04cCDWCLfat2//xSh3mzZtVF1d/bXt3F0LFy4sHTJkSFXd9kNNze3uOuOMM3b+8Y9//Mq9EX/5y186msUs6WvatWv3xTlSUlK++L5SUlK+qHfy5Mn9Fi5cWDp8+PC9Dz74YLfly5cfHerghxDPRy6flfRXSQPNrMzMflDftu6+XtJzkookvSzpBp6cAAAcjszMzOodO3akbtu2rc3evXvtlVdeOSYe5xk5cuTOmTNnZtQGlDfeeKOjJJ177rk7f/Ob36Tv3x8Z9a+9v+Coo446UHvfwtlnn/1ZYWFh53Xr1rWXpF27dqW8++677YcOHfp5WVlZu/Xr17eXpPnz56cdSY179uxJ6dOnz/6qqiqLPtbQoUN3z507t6skzZkz57DPEbeRBne/4hDrs+osT5c0PV71AACaVjwekTyY9u3b+6233lqen58/ODMzs+qkk076PB7nmTFjxtbJkyf3GTRoULa7W2ZmZtXSpUtLb7755sqSkpL2gwYNOjk1NdULCgoqb7/99sqCgoKPLrzwwv7du3ffv3LlypJZs2ZtnjBhwgn79u0zSZo2bdqH3/jGN6oeeuih90ePHn1SWlpa9Wmnnba7uLi4Y0NrnDp16tb8/PzBvXr12jd48OA9u3fvbiNJDz300AdXXnllvwcffPD4888//9POnTsf1i/oTI2NZoE3QqK5SsafTabGbr127dqVctRRR9WkpKRo9uzZXRcsWJC2ZMmSv9fdjqmxAQBo5d54441ON954Yx93V5cuXQ7MnTt38+HsT2gAAKCVGDVq1O6NGzc2eHrvpHhDFQAAiD9CAwAACIXQAAAAQiE0AACAULgREgAQFxOfnNCoU2M/c9X8Jn3vA76OkQYAABAKIw0NlIwvfAGAluy6667r1bdv331Tp06tlCJTXR999NEHXnrppWPT09P3FxUVdbrooos+yc3N3fvII49kVFVV2e9///u/n3zyyVXPPPPMMTNmzOixf//+lK5du1YvWLDgH717967eunVr6rhx4/p9+umnqUOHDt2zbNmyLqtXry7u0aNHdaK/30RgpAEAkBQmTZq0Y9GiRV/Mp/DCCy90TU9Pr96wYUPHRx999IPi4uL1Cxcu7FZSUtJh7dq1xVddddVHM2fO7C5J55133u533nlnQ3FxcdG4ceN23HPPPcdL0tSpU3ueddZZu4qKioovvfTST8rLy2POjtlaMNIAAEgKp59++t6PP/44dfPmzW3Ly8tTjznmmANZWVn7cnNzP+vbt+9+SerTp0/VhRde+E9JGjJkyN7a2R/fe++9dmPHjs2srKxsu2/fvpTevXtXSdKqVas6/+EPfyiVpHHjxu3s0qVLq55MkZEGAEDSuPjiiz956qmnuj799NNp3/ve93ZIX53uuu5U0rVTZE+ZMqXP9ddfX1FSUlL08MMPv19VVZUiRaayxpcIDQCApHHVVVftWLRoUdqLL77YddKkSZ+E3W/Xrl1t+vTps1+S5s6d2622PT8/f/eTTz6ZJknPP/98l507d7Zp/KpbDi5PAADiIhGPSObl5X3+2WefpWRkZOzr27fv/rVr13YIs98dd9yx9YorrjgxIyNjX15e3mdbtmxpL0WmwR43btwJ2dnZXYcPH747PT19/7HHHttqL1EQGgAASaWkpOSLCZlGjx69a/To0btql1etWrUx1rpJkyZ9OmnSpE/rHistLe3AihUrStq2bavXXnvtqDfeeOPojh07ttprFoQGAADqUVpa2u6yyy47saamRm3btvVZs2ZtTnRNiURoAACgHrm5uVXFxcUNnko62cTtRkgzm2NmFWa2Lqrtl2a2wczeNbPfm9mxUetuM7NSM9toZhfEqy4AQNzU1NTUWKKLwJEJ/g5rYq2L59MTcyWNqtP2qqQcd/+GpBJJt0mSmWVLmiDp5GCfR8ysVd+hCgAt0LrKyspjCA4tV01NjVVWVh4jaV2s9XG7POHuK8wsq07bn6IW/yZpXPD1JZLmu3uVpPfMrFRSvqS/xqs+AEDjqq6uvmbbtm2Pbdu2LUc80t9S1UhaV11dfU2slYm8p+H7kmoncOilSIioVRa0fY2ZTZY0WZL69OkTz/oAAIdh2LBhFZLGJLoOxE9CkqCZ3SGpWtLTtU0xNov5SIu7z3b3PHfPS09Pj1eJAACgjiYfaTCzAkmjJZ3jX76fs0xS76jNMiVtberaAABA/Zp0pMHMRkn6T0lj3H1P1KrFkiaYWXsz6yepv6RVTVkbAAA4uLiNNJjZs5LOlnScmZVJmqbI0xLtJb1qZpL0N3e/1t3Xm9lzkooUuWxxg7u32td0AgDQHMXz6YkrYjQ/fpDtp0uaHq96AADAkeGRGAAAEAqhAQAAhEJoAAAAoRAaAABAKIQGAAAQCqEBAACEQmgAAAChJHLCKgBxMOonCw69UQO9fO/lcTs2gOaPkQYAABAKoQEAAIRCaAAAAKEQGgAAQCiEBgAAEAqhAQAAhEJoAAAAoRAaAABAKIQGAAAQCqEBAACEErfQYGZzzKzCzNZFtaWZ2atmtin47Bq17jYzKzWzjWZ2QbzqAgAADRPPkYa5kkbVaZsqaYm795e0JFiWmWVLmiDp5GCfR8ysTRxrAwAAhyluocHdV0jaUaf5Eknzgq/nSRob1T7f3avc/T1JpZLy41UbAAA4fE19T0OGu5dLUvDZPWjvJemDqO3KgjYAANBMNJcbIS1Gm8fc0GyymRWaWWFlZWWcywIAALWaOjRsN7MekhR8VgTtZZJ6R22XKWlrrAO4+2x3z3P3vPT09LgWCwAAvtTUoWGxpILg6wJJL0S1TzCz9mbWT1J/SauauDYAAHAQqfE6sJk9K+lsSceZWZmkaZJmSHrOzH4gaYuk8ZLk7uvN7DlJRZKqJd3g7gfiVRsAADh8cQsN7n5FPavOqWf76ZKmx6seAABwZJrLjZAAAKCZIzQAAIBQCA0AACAUQgMAAAiF0AAAAEIhNAAAgFAIDQAAIBRCAwAACCVuL3cCgKZUmJcfnwNfeGt8jgu0QIw0AACAUEKFBjM7PUwbAABIXmFHGh4K2QYAAJLUQe9pMLPhkkZISjezW6JWdZHUJp6FAQCA5uVQN0K2k9Q52O7oqPadksbFqygAAND8HDQ0uPtyScvNbK67v99ENQEAgGYo7COX7c1stqSs6H3c/TvxKAoAADQ/YUPD7yT9RtJjkg7ErxwAANBchQ0N1e7+aFwrAQAAzVrYRy7/aGbXm1kPM0ur/RPXygAAQLMSdqShIPj8UVSbSzqhISc1s5slXRMcY62kf5XUSdICRe6b2CzpMnf/pCHHBwAAjS9UaHD3fo11QjPrJemHkrLdfa+ZPSdpgqRsSUvcfYaZTZU0VdJ/Hun5eB89AACNI1RoMLP/E6vd3Z84gvN2NLP9iowwbJV0m6Szg/XzJC1TI4QGAADQOMJenvhm1NcdJJ0j6S1Jhx0a3P1DM7tf0hZJeyX9yd3/ZGYZ7l4ebFNuZt1j7W9mkyVNlqQ+ffoc7ukBAEADhb088e/Ry2Z2jKQnG3JCM+sq6RJJ/SR9Kul3ZjYp7P7uPlvSbEnKy8vzhtQAAAAOX0Onxt4jqX8D9z1X0nvuXunu+yU9r8j8FtvNrIckBZ8VDTw+AACIg7D3NPxRkScdpMhEVYMlPdfAc26R9C0z66TI5YlzJBVK+kyRpzRmBJ8vNPD4AAAgDsLe03B/1NfVkt5397KGnNDdV5rZQkXuiaiW9LYilxs6S3rOzH6gSLAY35DjI37i9iSKxNMoANAChL2nYbmZZejLGyI3HclJ3X2apGl1mqsUGXUAAADNUKh7GszsMkmrFPnt/zJJK82MqbEBAGhFwl6euEPSN929QpLMLF3Sa5IWxqswAADQvIR9eiKlNjAEPj6MfQEAQBIIO9Lwspm9IunZYPlySS/FpyQAANAcHTQ0mNlJkjLc/UdmdqmkMySZpL9KeroJ6gMAAM3EoS4x/ErSLkly9+fd/RZ3v1mRUYZfxbc0AADQnBwqNGS5+7t1G929UJEprAEAQCtxqNDQ4SDrOjZmIQAAoHk7VGh408z+rW5j8NbG1fEpCQAANEeHenriJkm/N7Mr9WVIyJPUTtJ341gXAABoZg4aGtx9u6QRZjZSUk7Q/P/c/fW4VwYAAJqVsHNPLJW0NM61AACAZoy3OgIAgFAIDQAAIBRCAwAACIXQAAAAQiE0AACAUAgNAAAglISEBjM71swWmtkGMys2s+FmlmZmr5rZpuCzayJqAwAAsSVqpOHXkl5290GShkgqljRV0hJ37y9pSbAMAACaiSYPDWbWRdKZkh6XJHff5+6fSrpE0rxgs3mSxjZ1bQAAoH6JGGk4QVKlpN+a2dtm9piZHSUpw93LJSn47B5rZzObbGaFZlZYWVnZdFUDANDKJSI0pEo6VdKj7n6KpM90GJci3H22u+e5e156enq8agQAAHUkIjSUSSpz95XB8kJFQsR2M+shScFnRQJqAwAA9Wjy0ODu2yR9YGYDg6ZzJBVJWiypIGgrkPRCU9cGAADqF2qWyzj4d0lPm1k7Sf+Q9K+KBJjnzOwHkrZIGp+g2gAAQAwJCQ3u/o6kvBirzmniUgAAQEi8ERIAAIRCaAAAAKEQGgAAQCiEBgAAEAqhAQAAhEJoAAAAoRAaAABAKIQGAAAQCqEBAACEQmgAAAChEBoAAEAohAYAABBKoma5BFq9wrz8+Bz4wlvjc1wArR4jDQAAIBRCAwAACIXQAAAAQiE0AACAUAgNAAAglISFBjNrY2Zvm9mLwXKamb1qZpuCz66Jqg0AAHxdIkcabpRUHLU8VdISd+8vaUmwDAAAmomEhAYzy5T0L5Iei2q+RNK84Ot5ksY2cVkAAOAgEjXS8CtJP5ZUE9WW4e7lkhR8do+1o5lNNrNCMyusrKyMe6EAACCiyUODmY2WVOHuqxuyv7vPdvc8d89LT09v5OoAAEB9EvEa6dMljTGziyR1kNTFzJ6StN3Merh7uZn1kFSRgNoAAEA9mnykwd1vc/dMd8+SNEHS6+4+SdJiSQXBZgWSXmjq2gAAQP2a03saZkg6z8w2STovWAYAAM1EQme5dPdlkpYFX38s6ZxE1gMAAOrXnEYaAABAM0ZoAAAAoRAaAABAKIQGAAAQCqEBAACEQmgAAAChEBoAAEAohAYAABAKoQEAAIRCaAAAAKEQGgAAQCiEBgAAEAqhAQAAhEJoAAAAoRAaAABAKIQGAAAQCqEBAACEQmgAAAChNHloMLPeZrbUzIrNbL2Z3Ri0p5nZq2a2Kfjs2tS1AQCA+iVipKFa0q3uPljStyTdYGbZkqZKWuLu/SUtCZYBAEAz0eShwd3L3f2t4Otdkool9ZJ0iaR5wWbzJI1t6toAAED9EnpPg5llSTpF0kpJGe5eLkWChaTuCSwNAADUkbDQYGadJS2SdJO77zyM/SabWaGZFVZWVsavQAAA8BUJCQ1m1laRwPC0uz8fNG83sx7B+h6SKmLt6+6z3T3P3fPS09ObpmAAAJCQpydM0uOSit39v6NWLZZUEHxdIOmFpq4NAADULzUB5zxd0lWS1prZO0Hb7ZJmSHrOzH4gaYuk8QmoDQAA1KPJQ4O7/68kq2f1OU1ZCwAACI83QgIAgFAScXkCBzHxyQlxO/YzV82P27EBAMmPkQYAABAKoQEAAIRCaAAAAKEQGgAAQCiEBgAAEAqhAQAAhMIjl0hq8XqElcdXAbRGjDQAAIBQGGkAEBojN0DrxkgDAAAIhZEGAEgAXhmPloiRBgAAEAqhAQAAhEJoAAAAoRAaAABAKIQGAAAQCqEBAACE0uxCg5mNMrONZlZqZlMTXQ8AAIhoVqHBzNpI+r+SLpSULekKM8tObFUAAEBqZqFBUr6kUnf/h7vvkzRf0iUJrgkAAEgyd090DV8ws3GSRrn7NcHyVZJOc/cpUdtMljQ5WBwoaWOTF3r4jpP0UaKLSCL0Z+OiPxtPS+nLvu6enugi0PI0t9dIW4y2r6Qad58taXbTlNM4zKzQ3fMSXUeyoD8bF/3ZeOhLJLvmdnmiTFLvqOVMSVsTVAsAAIjS3ELDm5L6m1k/M2snaYKkxQmuCQAAqJldnnD3ajObIukVSW0kzXH39QkuqzG0qMspLQD92bjoz8ZDXyKpNasbIQEAQPPV3C5PAACAZorQAAAAQiE0HCEzu9nM1pvZOjN71sw6mFmamb1qZpuCz6717Msrs6OY2Y1BP643s5uCtrvM7EMzeyf4c1E9+7b6vjSzOWZWYWbrYqz7DzNzMzsuqu22oL82mtkF9Rwz1M9ysjKzNmb2tpm9GCzX2x/0J1oDQsMRMLNekn4oKc/dcxS5eXOCpKmSlrh7f0lLguW6+/LK7ChmliPp3xR5K+gQSaPNrH+w+gF3Hxr8eSnGvvRlxFxJo+o2mllvSedJ2hLVlq3Iz+rJwT6PBP1Y1yF/lpPcjZKKo5Zj9gf9idaC0HDkUiV1NLNUSZ0Uea/EJZLmBevnSRobYz9emf1VgyX9zd33uHu1pOWSvhtyX/pSkruvkLQjxqoHJP1YX31R2iWS5rt7lbu/J6lUkX6sK8zPclIys0xJ/yLpsajm+vqD/kSrQGg4Au7+oaT7FfkNrlzSP939T5Iy3L082KZcUvcYu/eS9EHUclnQ1lqtk3SmmXUzs06SLtKXL/qaYmbvBsPvsYZz6ct6mNkYSR+6+5o6q8L2WZif5WT1K0XCVk1UW339QX+iVSA0HIHgH7BLJPWT1FPSUWY2KezuMdpa7fOv7l4s6ReSXpX0sqQ1kqolPSrpRElDFQlmM2PsTl/GEISvOyT9NNbqGG2tvs9qmdloSRXuvjrsLjHa6E8kHULDkTlX0nvuXunu+yU9L2mEpO1m1kOSgs+KGPvyyuw63P1xdz/V3c9UZJh9k7tvd/cD7l4j6X8Ue8iXvoztREUC7Roz26xIv7xlZscrfJ+F+VlORqdLGhP023xJ3zGzp1R/f9CfaBUIDUdmi6RvmVknMzNJ5yhy09RiSQXBNgWSXoixL6/MrsPMugeffSRdKunZ2v/BBr6ryGWMuujLGNx9rbt3d/csd89S5B+2U919myL9M8HM2ptZP0n9Ja2KcZgwP8tJx91vc/fMoN8mSHrd3Sep/v6gP9EqEBqOgLuvlLRQ0luS1irSn7MlzZB0npltUuSu9RmSZGY9zeylYN9qSbWvzC6W9FySvDL7SCwysyJJf5R0g7t/Ium/zGytmb0raaSkmyX6MhYze1bSXyUNNLMyM/tBfdsG/fOcpCJFLgfd4O4HguM8Zma1MzXG/FluxWL2B/2J1oLXSAMAgFAYaQAAAKEQGgAAQCiEBgAAEAqhAQAAhEJoAAAAoRAakFTM7I5glsx3g1kxT4vz+ZZFPU4XZvu5Zvaema0xsxIzeyKY+Kyh5x8bPTnX4dYDAIeD0ICkYWbDJY1W5AVG31DkjZ0fHHyvhPiRuw+RNFDS25KWBi+laoixiszsCQBxR2hAMukh6SN3r5Ikd//I3bdKkpn91MzeNLN1ZjY7eINn7W/mD5jZCjMrNrNvmtnzZrbJzO4Ltskysw1mNi8YwVgYzOvwFWZ2vpn91czeMrPfmVnngxXrEQ9I2qbItN71HsPMNpvZL8xsVfDnJDMbIWmMpF8GoyonBoceH2xTYmbfboyOBQCJ0IDk8idJvYN/LB8xs7Oi1j3s7t909xxJHRUZkai1L5jv4jeKvNb3Bkk5kq42s27BNgMlzQ5GMHZKuj76xGZ2nKQ7JZ3r7qdKKpR0S8i635I0KMQxdrp7vqSHJf3K3f+iyGuJf+TuQ93978F2qcF2N0maFrIGADgkQgOShrvvljRM0mRJlZIWmNnVweqRZrbSzNZK+o6kk6N2rZ2nYq2k9e5eHoxW/ENfTkL0gbu/EXz9lKQz6pz+W4pcJnjDzN5RZF6BviFLr50h8VDHeDbqc/hBjvd88LlaUlbIGgDgkFITXQDQmIL3/S+TtCwICAVmNl/SI5Ly3P0DM7tLUoeo3aqCz5qor2uXa/8bqfu+9brLJulVd7+iAWWfImlJiGN4PV/XVfs9HBD/jQNoRIw0IGmY2UAz6x/VNFTS+/oyIHwU3CMwrgGH7xPcaClJV0j63zrr/ybpdDM7Kailk5kNOES9ZmY/VORejJdDHOPyqM+/Bl/vknR0A74fADhshAYkk86S5plZUTArZraku9z9U0n/o8jlhz8oMpX24SpWZNTiXUlpkh6NXunulZKuVmQ673cVCQCD6jnWL81sjaQSSd+UNNLd94U4RnszWynpRgWzfUqaL+lHZvZ21I2QABAXzHIJHIKZZUl6MbiJMlE1bFbk8spHiaoBABhpAAAAoTDSAAAAQmGkAQAAhEJoAAAAoRAaAABAKIQGAAAQCqEBAACE8v8Bh8uhL2VpakgAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.barplot(x='sample_depth', y='count', hue='virus_type', data=virus_count, palette='Set1')\n",
    "plt.legend(title='', bbox_to_anchor=(1.05, 1), loc='upper left')\n",
    "plt.xlabel('Sample Depth')\n",
    "plt.ylabel('Count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f62533a9-7fd7-4a10-a31d-ead7c0db360e",
   "metadata": {},
   "source": [
    "## Evaluating the Strength of Host-Virus Associations\n",
    "\n",
    "### Plots: Distribution of adjusted_inter_vs_intra_ratio "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd5dc9d6-39f2-41ff-8a50-c0548aaf69d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# each data frame is created by grouping vmag_assoc by sample_name and virus_type and getting the group corresponding to that data frame\n",
    "\n",
    "jv119_contig = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV119', 'contig'))\n",
    "jv119_vMAG = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV119', 'vmag'))\n",
    "jv119_uninfect = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV119', 'uninfected mag'))\n",
    "\n",
    "\n",
    "jv121_contig = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV121', 'contig'))\n",
    "jv121_vMAG = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV121', 'vmag'))\n",
    "\n",
    "jv132_contig = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV132', 'contig'))\n",
    "jv132_vMAG = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV132', 'vmag'))\n",
    "\n",
    "jv154_contig = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV154', 'contig'))\n",
    "jv154_vMAG = proximeta.groupby(['sample_name', 'virus_type'], as_index=False).get_group(('JV154', 'vmag'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a32575f2-cbaa-49c4-ac66-7d5d72dde094",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot contigs\n",
    "\n",
    "fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(10,8), sharex=True, sharey=True)\n",
    "\n",
    "palette = sns.color_palette('gist_earth')\n",
    "\n",
    "sns.histplot(jv119_contig['adjusted_inter_vs_intra_ratio'],ax= axes[0,0], bins=10, color=palette[0], edgecolor='black')\n",
    "axes[0,0].set_xlabel('Adjusted Inter vs Intra Ratio')\n",
    "axes[0,0].set_ylabel('Frquency')\n",
    "axes[0,0].set_title('JV119 Contigs (Sample Depth 400 Meters)')\n",
    "axes[0,0].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv154_contig['adjusted_inter_vs_intra_ratio'],ax= axes[1,1], bins=10, color=palette[3], edgecolor='black')\n",
    "axes[0,1].set_xlabel('Adjusted Inter vs Intra Ratio')\n",
    "axes[0,1].set_ylabel('Frquency')\n",
    "axes[0,1].set_title('JV154 Contigs (Sample Depth 140 Meters)')\n",
    "axes[0,1].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv121_contig['adjusted_inter_vs_intra_ratio'],ax= axes[0,1], bins=10, color=palette[1], edgecolor='black')\n",
    "axes[1,0].set_xlabel('Adjusted Inter vs Intra Ratio')\n",
    "axes[1,0].set_ylabel('Frquency')\n",
    "axes[1,0].set_title('JV121 Contigs (Sample Depth 95 Meters)')\n",
    "axes[1,0].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv132_contig['adjusted_inter_vs_intra_ratio'],ax= axes[1,0], bins=10, color=palette[2], edgecolor='black')\n",
    "axes[1,1].set_xlabel('Adjusted Inter vs Intra Ratio')\n",
    "axes[1,1].set_ylabel('Frquency')\n",
    "axes[1,1].set_title('JV132 Contigs (Sample Depth 80 Meters)')\n",
    "axes[1,1].set_yscale('log')\n",
    "\n",
    "fig.suptitle('Distribution of the Inter vs Intra Ratio',fontsize=16)\n",
    "plt.tight_layout() # minimize white space in figure"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2dabafc-90aa-4ed0-b565-a6b4cc6bd636",
   "metadata": {},
   "source": [
    "This set of plots shows the distribution of the adjusted_inter_vs_intra_ratio column for the contigs for all four samples. jv119 is on the top left, jv154 is on the top right, jv121 is on the bottom left, and jv132 is on the bottom right. The x-axes represent values of the inter vs intra ratio and the y-axes represent the frequency of each ratio value in the dataset.\n",
    "\n",
    "There is evidence for a bimodal distributions for the samples jv119, jv154, and jv132. As depth generally decreases, it appears that this ratio is concentrated towards lower values.\n",
    "\n",
    "A value close to one signals a 'real' infection between the host and the virus. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9006abd-b7f9-4895-9619-7bb888e8ae24",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot all vmags\n",
    "\n",
    "fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(10,8), sharex=True, sharey=True)\n",
    "\n",
    "palette = sns.color_palette('gist_earth')\n",
    "\n",
    "sns.histplot(jv119_vMAG['adjusted_inter_vs_intra_ratio'],ax= axes[0,0], bins=10, color=palette[0], edgecolor='black')\n",
    "axes[0,0].set_xlabel('Adjusted Inter vs Intra Ratio')\n",
    "axes[0,0].set_ylabel('Frquency')\n",
    "axes[0,0].set_title('JV119 vMAGs (Sample Depth 400 Meters)')\n",
    "\n",
    "sns.histplot(jv154_vMAG['adjusted_inter_vs_intra_ratio'],ax= axes[1,1], bins=10, color=palette[1], edgecolor='black')\n",
    "axes[0,1].set_xlabel('Adjusted Inter vs Intra Ratio')\n",
    "axes[0,1].set_ylabel('Frquency')\n",
    "axes[0,1].set_title('JV154 vMAGs (Sample Depth 140 Meters)')\n",
    "\n",
    "sns.histplot(jv121_vMAG['adjusted_inter_vs_intra_ratio'],ax= axes[0,1], bins=10, color=palette[2], edgecolor='black')\n",
    "axes[1,0].set_xlabel('Adjusted Inter vs Intra Ratio')\n",
    "axes[1,0].set_ylabel('Frquency')\n",
    "axes[1,0].set_title('JV121 vMAGs (Sample Depth 95 Meters)')\n",
    "\n",
    "sns.histplot(jv132_vMAG['adjusted_inter_vs_intra_ratio'],ax= axes[1,0], bins=10, color=palette[3], edgecolor='black')\n",
    "axes[1,1].set_xlabel('Adjusted Inter vs Intra Ratio')\n",
    "axes[1,1].set_ylabel('Frquency')\n",
    "axes[1,1].set_title('JV132 vMAGs (Sample Depth 80 Meters)')\n",
    "\n",
    "fig.suptitle('Distribution of the Inter vs Intra Ratio',fontsize=16)\n",
    "plt.tight_layout()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e0a438b-2af9-4634-84cd-a079f59a87a0",
   "metadata": {},
   "source": [
    "This set of plots shows the distribution of the adjusted_inter_vs_intra_ratio column for the vmags for all four samples. jv119 is on the top left, jv154 is on the top right, jv121 is on the bottom left, and jv132 is on the bottom right. The x-axes represent values of the inter vs intra ratio and the y-axes represent the frequency of each ratio value in the dataset. In general, there are a lot more contigs than vmags, so these plots are generated with less data, making it hard to make definitive conclusions from these plots alone.\n",
    "\n",
    "There is could be some evidence for a bimodal distributions for all four samples.\n",
    "\n",
    "A value close to one signals a 'real' infection between the host and the virus. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "563a9743-dc61-4812-afaa-fd6447b0d33c",
   "metadata": {},
   "source": [
    "### Plots: Distribution of adjusted_inter_connective_linkage_density"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b991528-5a00-46de-b2e5-a53e4ba4d137",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot contigs\n",
    "\n",
    "fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(10,8), sharex=True, sharey=True)\n",
    "\n",
    "palette = sns.color_palette('gist_earth')\n",
    "\n",
    "sns.histplot(jv119_contig['adjusted_inter_linkage_density'],ax= axes[0,0], bins=10, color=palette[0], edgecolor='black')\n",
    "axes[0,0].set_xlabel('Adjusted Inter Connective Linkage Density (reads/kbp^2)')\n",
    "axes[0,0].set_ylabel('Frquency')\n",
    "axes[0,0].set_title('JV119 Contigs (Sample Depth 400 Meters)')\n",
    "axes[0,0].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv154_contig['adjusted_inter_linkage_density'],ax= axes[1,1], bins=10, color=palette[1], edgecolor='black')\n",
    "axes[0,1].set_xlabel('Adjusted Inter Connective Linkage Density (reads/kbp^2)')\n",
    "axes[0,1].set_ylabel('Frquency')\n",
    "axes[0,1].set_title('JV154 Contigs (Sample Depth 140 Meters)')\n",
    "axes[0,1].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv121_contig['adjusted_inter_linkage_density'],ax= axes[0,1], bins=10, color=palette[2], edgecolor='black')\n",
    "axes[1,0].set_xlabel('Adjusted Inter Connective Linkage Density (reads/kbp^2)')\n",
    "axes[1,0].set_ylabel('Frquency')\n",
    "axes[1,0].set_title('JV121 Contigs (Sample Depth 95 Meters)')\n",
    "axes[1,0].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv132_contig['adjusted_inter_linkage_density'],ax= axes[1,0], bins=10, color=palette[3], edgecolor='black')\n",
    "axes[1,1].set_xlabel('Adjusted Inter Connective Linkage Density (reads/kbp^2)')\n",
    "axes[1,1].set_ylabel('Frquency')\n",
    "axes[1,1].set_title('JV132 Contigs (Sample Depth 80 Meters)')\n",
    "axes[1,1].set_yscale('log')\n",
    "\n",
    "fig.suptitle('Distribution of the Inter Linkage Density',fontsize=16)\n",
    "plt.tight_layout()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95e6fdf9-c2f1-4aa0-9ae4-d534e58d93ac",
   "metadata": {},
   "source": [
    "This set of plots shows the distribution of the adjusted_inter_connective_linkage_density column for the contigs for all four samples. jv119 is on the top left, jv154 is on the top right, jv121 is on the bottom left, and jv132 is on the bottom right. The x-axes represent values of the inter connective linkage density and the y-axes represent the frequency of each density value in the dataset.\n",
    "\n",
    "There is could be some evidence for a bimodal distributions for all four samples.\n",
    "\n",
    "A higher value means more linkage between the host and the virus."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e8fe526-45da-4c61-a958-5729c19fb7d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot all vMAGs together\n",
    "\n",
    "fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(10,8), sharex=True, sharey=True)\n",
    "\n",
    "palette = sns.color_palette('gist_earth')\n",
    "\n",
    "sns.histplot(jv119_vMAG['adjusted_inter_linkage_density'], bins=10, color=palette[0], edgecolor='black', ax= axes[0,0])\n",
    "axes[0,0].set_xlabel('Adjusted Inter Connective Linkage Density (reads/kbp^2)')\n",
    "axes[0,0].set_ylabel('Frquency')\n",
    "axes[0,0].set_title('JV119 vMAGs (Sample Depth 400 Meters)')\n",
    "axes[0,0].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv154_vMAG['adjusted_inter_linkage_density'], bins=10, color=palette[1], edgecolor='black', ax= axes[1,1])\n",
    "axes[1,1].set_xlabel('Adjusted Inter Connective Linkage Density (reads/kbp^2)')\n",
    "axes[1,1].set_ylabel('Frquency')\n",
    "axes[1,1].set_title('JV154 vMAGs (Sample Depth 140 Meters)')\n",
    "axes[1,1].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv121_vMAG['adjusted_inter_linkage_density'], bins=10, color=palette[2], edgecolor='black', ax= axes[0,1])\n",
    "axes[0,1].set_xlabel('Adjusted Inter Connective Linkage Density (reads/kbp^2)')\n",
    "axes[0,1].set_ylabel('Frquency')\n",
    "axes[0,1].set_title('JV121 vMAGs (Sample Depth 95 Meters)')\n",
    "axes[0,1].set_yscale('log')\n",
    "\n",
    "sns.histplot(jv132_vMAG['adjusted_inter_linkage_density'], bins=10, color=palette[3], edgecolor='black', ax= axes[1,0])\n",
    "axes[1,0].set_xlabel('Adjusted Inter Connective Linkage Density (reads/kbp^2)')\n",
    "axes[1,0].set_ylabel('Frquency')\n",
    "axes[1,0].set_title('JV132 vMAGs (Sample Depth 80 Meters)')\n",
    "axes[1,0].set_yscale('log')\n",
    "\n",
    "fig.suptitle('Distribution of the Inter Linkage Density',fontsize=16)\n",
    "plt.tight_layout()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f100d55-8b48-4af5-b71a-3b5c89d22323",
   "metadata": {},
   "source": [
    "This set of plots shows the distribution of the adjusted_inter_connective_linkage_density column for the contigs for all four samples. jv119 is on the top left, jv154 is on the top right, jv121 is on the bottom left, and jv132 is on the bottom right. The x-axes represent values of the inter connective linkage density and the y-axes represent the frequency of each density value in the dataset. In general, there are a lot more contigs than vmags, so these plots are generated with less data, making it hard to make definitive conclusions from these plots alone.\n",
    "\n",
    "There is could be some evidence for a bimodal distributions for all four samples.\n",
    "\n",
    "A higher value means more linkage between the host and the virus."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3dc78422-e34b-472f-9b2d-a6ad508ced64",
   "metadata": {},
   "source": [
    "***NEXT STEP: ARE THERE PHYLOGENY PATTERNS ASSOCIATED WITH THESE METRICS? ARE THERE ONLY CERTAIN HOST TAXA BELONGING TO HIGHER OR LOWER VALUES?***"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9932596-4f40-4a01-b469-0746593853b1",
   "metadata": {},
   "source": [
    "## Viral Copies per Cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "670e0b50-15ce-4bdf-9c8d-bb2089c79f17",
   "metadata": {},
   "outputs": [],
   "source": [
    "viruses = pd.concat([proximeta.groupby('virus_type').get_group('contig'),proximeta.groupby('virus_type').get_group('vmag')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c51b989e-5978-4feb-9fa1-01191ceaeeed",
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a jitterplot of vmag_assoc data frame with sample_depth on the x-axis, viral_copies_per_cell on the y-axis, and the data points colored by virus_type\n",
    "sns.stripplot(x='sample_depth', y='viral_copies_per_cell', data=viruses, jitter=True, hue='virus_type',palette=sns.color_palette('Set1'))\n",
    "plt.xlabel('Sample Depth')\n",
    "plt.ylabel('Number of Viral Copies per Cell')\n",
    "plt.legend(bbox_to_anchor=(1.4, 1.05),title='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac18b196-59a3-4767-a2c0-661f683f2325",
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.boxplot(x='sample_depth', y='viral_copies_per_cell', data=viruses, hue='virus_type',palette=sns.color_palette('Set1'))\n",
    "plt.xlabel('Sample Depth')\n",
    "plt.ylabel('Number of Viral Copies per Cell')\n",
    "plt.legend(bbox_to_anchor=(1.4, 1.05),title='')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "645ef549-0521-482f-aaf0-9adb1649fc63",
   "metadata": {},
   "source": [
    "## Number of Associated Viruses and the Total Number of vSAGs/vMAGs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2d422ec-6a56-43ab-8b7b-7764373c5fcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# creata a data frame by grouping vmag_assoc by virus_type and getting the group vMAG\n",
    "vmags = vmag_assoc.groupby('virus_type', as_index=False).get_group('vMAG')\n",
    "\n",
    "# create a list of sample depths\n",
    "sample_depth = [80,95,140,400]\n",
    "\n",
    "# create a data frame that counts the number of vMAGs for each sample depth\n",
    "vmag_assoc_count = pd.DataFrame(vmags['sample_depth'].value_counts())\n",
    "\n",
    "# rename the column sample_depth to count\n",
    "vmag_assoc_count.rename(columns={'sample_depth': 'count'}, inplace=True)\n",
    "\n",
    "# create a sample_depth column using the sample_depth list\n",
    "vmag_assoc_count['sample_depth'] = sample_depth\n",
    "\n",
    "# create a column named type\n",
    "vmag_assoc_count['type'] = 'vMAGs connected to a Host'\n",
    "\n",
    "# create a data frame that counts the number of total vmags using the length of each sample's dataframe\n",
    "vmag_tot = pd.DataFrame([len(jv119_vmag_sum), len(jv121_vmag_sum), len(jv132_vmag_sum), len(jv154_vmag_sum)])\n",
    "\n",
    "# rename first column\n",
    "vmag_tot.rename(columns={0: 'count'}, inplace=True)\n",
    "\n",
    "# merge sample_depth list as a data frame and vmag_tot \n",
    "vmag_tot = pd.DataFrame(pd.concat([vmag_tot,pd.DataFrame(sample_depth)],axis=1))\n",
    "\n",
    "# rename first column\n",
    "vmag_tot.rename(columns={0: 'sample_depth'}, inplace=True)\n",
    "\n",
    "# create a column named type\n",
    "vmag_tot['type'] = 'Total vMAGs'\n",
    "\n",
    "# merge vmag_assoc_count and vmag_tot data frames\n",
    "combo1 = pd.concat([vmag_assoc_count,vmag_tot])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bda60c91-834c-4c55-a897-43c907a1b224",
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(12, 6), sharex=True, sharey=True)\n",
    "\n",
    "sns.barplot(x='sample_depth', y='count', data=len_vmag_df, color='royalblue', ax=axes[0])\n",
    "axes[0].set_xlabel('Sample Depth')\n",
    "axes[0].set_ylabel('Number of Associated vMAGs')\n",
    "axes[0].set_title('Number of Associated vMAGs per Sample')\n",
    "\n",
    "sns.barplot(x='sample_depth', y='count', data=len_vmag_df, color='crimson', ax=axes[1])\n",
    "axes[1].set_xlabel('Sample Depth')\n",
    "axes[1].set_ylabel('Number of Total Viral MAGs')\n",
    "axes[1].set_title('Number of Total Viral MAGs per Sample')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7d5e1f0-7f4a-45d5-b738-76de2f7e781e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a custom color palette\n",
    "custom_palette1 = ['crimson','royalblue']\n",
    "\n",
    "# create a barplot of combo1 with sample_depth on the x-axis and count on the y-axis colored by the type column using the custom palette1\n",
    "sns.barplot(x=\"sample_depth\", y=\"count\", hue=\"type\", data=combo1, palette=custom_palette1)\n",
    "plt.xlabel('Sample Depth')\n",
    "plt.ylabel('Count')\n",
    "plt.legend(title='',loc='upper left', bbox_to_anchor=(1, 1)) # set legend title and location outside of the plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2c48086-2620-4682-bed5-3d218b18f7ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a list of sample depths\n",
    "sample_depth = [80,95,140,400]\n",
    "\n",
    "# create a data frame counting how many sags at each depth\n",
    "sag_assoc_count = pd.DataFrame(sag_sum['depth'].value_counts())\n",
    "\n",
    "# rename depth column to count\n",
    "sag_assoc_count.rename(columns={'depth': 'count'}, inplace=True)\n",
    "\n",
    "# create a sample_depth column using sample_depth list\n",
    "sag_assoc_count['sample_depth'] = sample_depth\n",
    "\n",
    "# create a column named type\n",
    "sag_assoc_count['type'] = 'vSAGs connected to a Host'\n",
    "\n",
    "# create a data frame the counts the number of total sags at each depth\n",
    "vsag_tot = pd.DataFrame([len(sag_gtdb[sag_gtdb['depth'] == 80]), len(sag_gtdb[sag_gtdb['depth'] == 95]), len(sag_gtdb[sag_gtdb['depth'] == 140]), len(sag_gtdb[sag_gtdb['depth'] == 400])])\n",
    "\n",
    "# rename first column\n",
    "vsag_tot.rename(columns={0: 'count'}, inplace=True)\n",
    "\n",
    "# add sample depth column using the sample_depth list\n",
    "vsag_tot = pd.DataFrame(pd.concat([vsag_tot,pd.DataFrame(sample_depth)],axis=1))\n",
    "\n",
    "# rename fist column\n",
    "vsag_tot.rename(columns={0: 'sample_depth'}, inplace=True)\n",
    "\n",
    "# create a column named type\n",
    "vsag_tot['type'] = 'Total vSAGs'\n",
    "\n",
    "# merge sag_assoc_count and vsag_tot data frames\n",
    "combo2 = pd.concat([sag_assoc_count,vsag_tot])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "498d54d6-3c77-4084-af4c-b4ff84de114a",
   "metadata": {},
   "outputs": [],
   "source": [
    "custom_palette2 = ['goldenrod','forestgreen']\n",
    "\n",
    "sns.barplot(x=\"sample_depth\", y=\"count\", hue=\"type\", data=combo2, palette=custom_palette2)\n",
    "plt.xlabel('Sample Depth')\n",
    "plt.ylabel('Count')\n",
    "plt.legend(title='',loc='upper left', bbox_to_anchor=(1, 1))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
