{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "f399c268-2769-44d7-8922-0e703ee8509c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load packages\n",
    "import pandas as pd\n",
    "import os\n",
    "import os.path as op\n",
    "import sys\n",
    "import csv\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import glob\n",
    "import seaborn as sns\n",
    "from collections import Counter\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# file paths\n",
    "path = '/Users/melissaherring/Google Drive/My Drive/MH_project/'\n",
    "#path = '/Users/juliabrown/Google Drive/My Drive/projects/OMZvir_round2/MH_project/'\n",
    "\n",
    "# define functions \n",
    "def split_classification(df):\n",
    "    df[['domain','phyla','class','order','family','genus','species']] = df.classification.str.split(';', expand=True)\n",
    "    df['domain'] = df['domain'].str.replace('d__', '')\n",
    "    df['phyla'] = df['phyla'].str.replace('p__', '')\n",
    "    df['class'] = df['class'].str.replace('c__', '')\n",
    "    df['order'] = df['order'].str.replace('o__', '')\n",
    "    df['family'] = df['family'].str.replace('f__', '')\n",
    "    df['genus'] = df['genus'].str.replace('g__', '')\n",
    "    df['species'] = df['species'].str.replace('s__', '')\n",
    "    return df\n",
    "\n",
    "\n",
    "''' JOIN ASSOCIATED CONTIG AND VMAG DATA '''\n",
    "\n",
    "\n",
    "# input contig data\n",
    "jv119_contig = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-119_proximeta/viral_hosts/viral_host_associations_filtered.tsv'), sep ='\\t')\n",
    "jv121_contig = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-121_proximeta/viral_hosts/viral_host_associations_filtered.tsv'), sep ='\\t')\n",
    "jv132_contig = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-132_proximeta/viral_hosts/viral_host_associations_filtered.tsv'), sep ='\\t')\n",
    "jv154_contig = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-154_proximeta/viral_hosts/viral_host_associations_filtered.tsv'), sep ='\\t')\n",
    "\n",
    "# create a list of all contig dataframes\n",
    "contig_list = [jv119_contig, jv121_contig, jv132_contig, jv154_contig]\n",
    "\n",
    "## input vMAG data\n",
    "jv119_vmag = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-119_proximeta/vMag_hosts/vMAG_host_associations_filtered.tsv'), sep ='\\t')\n",
    "jv121_vmag = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-121_proximeta/vMag_hosts/vMAG_host_associations_filtered.tsv'), sep ='\\t')\n",
    "jv132_vmag = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-132_proximeta/vMag_hosts/vMAG_host_associations_filtered.tsv'), sep ='\\t')\n",
    "jv154_vmag = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-154_proximeta/vMag_hosts/vMAG_host_associations_filtered.tsv'), sep ='\\t')\n",
    "\n",
    "# create a list of all vmag dataframes\n",
    "vmag_list = [jv119_vmag, jv121_vmag, jv132_vmag, jv154_vmag]\n",
    "\n",
    "# create virus_type column for each dataframe\n",
    "for df in contig_list:\n",
    "    df['virus_type'] = 'contig'\n",
    "for df in vmag_list:\n",
    "    df['virus_type'] = 'vmag'\n",
    "\n",
    "# create sample_name and sample_depth column for each dataframe\n",
    "for df in list([jv119_contig, jv119_vmag]):\n",
    "    df['sample_name'] = 'JV119'\n",
    "    df['sample_depth'] = 400\n",
    "for df in list([jv121_contig, jv121_vmag]):\n",
    "    df['sample_name'] = 'JV121'\n",
    "    df['sample_depth'] = 95\n",
    "for df in list([jv132_contig, jv132_vmag]):\n",
    "    df['sample_name'] = 'JV132'\n",
    "    df['sample_depth'] = 80\n",
    "for df in list([jv154_contig, jv154_vmag]):\n",
    "    df['sample_name'] = 'JV154'\n",
    "    df['sample_depth'] = 140\n",
    "\n",
    "# rename columns to match betweeen contig and vmag dataframes\n",
    "\n",
    "## create dictionaries to use for column mapping -> old_col_name : new_col_name\n",
    "column_mapping_contig = {\n",
    "    'mobile_contig_name': 'virus_name',\n",
    "    'mobile_contig_length (bp)': 'virus_length',\n",
    "    'mobile_contig_read_count (reads)': 'virus_read_count',\n",
    "    'mobile_contig_read_depth (reads/kbp)': 'virus_read_depth',\n",
    "    'mobile_contig_read_depth_in_this_cluster (reads/kbp)': 'virus_read_depth_in_host',\n",
    "    'cluster_name': 'host_name',\n",
    "    'cluster_length (bp)': 'host_length',\n",
    "    'cluster_read_count (reads)': 'host_read_count',\n",
    "    'cluster_read_depth (reads/kbp)': 'host_read_depth',\n",
    "    'intra_read_count (reads)': 'intra_read_count',\n",
    "    'intra_linkage_density (reads/kbp^2)': 'intra_linkage_density',\n",
    "    'inter_read_count (reads)': 'inter_read_count',\n",
    "    'raw_inter_linkage_density (reads/kbp^2)': 'raw_inter_linkage_density',\n",
    "    'raw_inter_vs_intra_ratio': 'raw_inter_vs_intra_ratio',\n",
    "    'mobile_element_copies_per_cell': 'viral_copies_per_cell',\n",
    "    'adjusted_inter_connective_linkage_density (reads/kbp^2)': 'adjusted_inter_linkage_density',\n",
    "    'adjusted_inter_vs_intra_ratio': 'adjusted_inter_vs_intra_ratio',\n",
    "    'sample_name':'sample_name',\n",
    "    'virus_type': 'virus_type',\n",
    "    'sample_depth': 'sample_depth'\n",
    "}\n",
    "column_mapping_vmag = {\n",
    "    'mobile_cluster_name': 'virus_name',\n",
    "    'mobile_cluster_length (bp)': 'virus_length',\n",
    "    'mobile_cluster_read_count (reads)': 'virus_read_count',\n",
    "    'mobile_cluster_read_depth (reads/kbp)': 'virus_read_depth',\n",
    "    'mobile_cluster_read_depth_in_this_cluster (reads/kbp)': 'virus_read_depth_in_host',\n",
    "    'cluster_name': 'host_name',\n",
    "    'cluster_length (bp)': 'host_length',\n",
    "    'cluster_read_count (reads)': 'host_read_count',\n",
    "    'cluster_read_depth (reads/kbp)': 'host_read_depth',\n",
    "    'intra_read_count (reads)': 'intra_read_count',\n",
    "    'intra_linkage_density (reads/kbp^2)': 'intra_linkage_density',\n",
    "    'inter_read_count (reads)': 'inter_read_count',\n",
    "    'raw_inter_linkage_density (reads/kbp^2)': 'raw_inter_linkage_density',\n",
    "    'raw_inter_vs_intra_ratio': 'raw_inter_vs_intra_ratio',\n",
    "    'mobile_element_copies_per_cell': 'viral_copies_per_cell',\n",
    "    'adjusted_inter_connective_linkage_density (reads/kbp^2)': 'adjusted_inter_linkage_density',\n",
    "    'adjusted_inter_vs_intra_ratio': 'adjusted_inter_vs_intra_ratio',\n",
    "    'sample_name':'sample_name',\n",
    "    'virus_type': 'virus_type',\n",
    "    'sample_depth': 'sample_depth'\n",
    "}\n",
    "\n",
    "## for loops to rename columns\n",
    "for df in contig_list:\n",
    "    df.rename(columns=column_mapping_contig, inplace=True)\n",
    "for df in vmag_list: \n",
    "    df.rename(columns=column_mapping_vmag, inplace=True)\n",
    "\n",
    "# merge all contig and vmag dataframes together\n",
    "proximeta = pd.concat([jv119_contig, jv121_contig, jv132_contig, jv154_contig, jv119_vmag, jv121_vmag, jv132_vmag, jv154_vmag])\n",
    "\n",
    "\n",
    "''' MERGE MAG GTDB DATA TOGETHER '''\n",
    "\n",
    "\n",
    "# input mag gtdb files\n",
    "jv119_arc = pd.read_csv(op.join(path,'data/mag_data/jv-119_gtdbtk/jv-119.ar53.summary.tsv'),sep='\\t')\n",
    "jv119_bac = pd.read_csv(op.join(path,'data/mag_data/jv-119_gtdbtk/jv-119.bac120.summary.tsv'),sep='\\t')\n",
    "jv121_arc = pd.read_csv(op.join(path,'data/mag_data/jv-121_gtdbtk/jv-121.ar53.summary.tsv'),sep='\\t')\n",
    "jv121_bac = pd.read_csv(op.join(path,'data/mag_data/jv-121_gtdbtk/jv-121.bac120.summary.tsv'),sep='\\t')\n",
    "jv132_arc = pd.read_csv(op.join(path,'data/mag_data/jv-132_gtdbtk/jv-132.ar53.summary.tsv'),sep='\\t')\n",
    "jv132_bac = pd.read_csv(op.join(path,'data/mag_data/jv-132_gtdbtk/jv-132.bac120.summary.tsv'),sep='\\t')\n",
    "jv154_arc = pd.read_csv(op.join(path,'data/mag_data/jv-154_gtdbtk/jv-154.ar53.summary.tsv'),sep='\\t')\n",
    "jv154_bac = pd.read_csv(op.join(path,'data/mag_data/jv-154_gtdbtk/jv-154.bac120.summary.tsv'),sep='\\t')\n",
    "\n",
    "# create sample_name and sample_depth column for each mag gtdb dataframe\n",
    "for df in list([jv119_arc, jv119_bac]):\n",
    "    df['sample_name'] = 'JV119'\n",
    "    df['sample_depth'] = 400\n",
    "for df in list([jv121_arc, jv121_bac]):\n",
    "    df['sample_name'] = 'JV121'\n",
    "    df['sample_depth'] = 95\n",
    "for df in list([jv132_arc, jv132_bac]):\n",
    "    df['sample_name'] = 'JV132'\n",
    "    df['sample_depth'] = 80\n",
    "for df in list([jv154_arc, jv154_bac]):\n",
    "    df['sample_name'] = 'JV154'\n",
    "    df['sample_depth'] = 140\n",
    "    \n",
    "# combine all mag gtdb dataframes\n",
    "mag_gtdb = pd.concat([jv119_arc, jv119_bac, jv121_arc, jv121_bac, jv132_arc, jv132_bac, jv154_arc, jv154_bac])\n",
    "\n",
    "# split the classification into tax level columns and rename\n",
    "mag_gtdb = split_classification(mag_gtdb)\n",
    "\n",
    "\n",
    "''' MERGE PROXIMETA AND GTDB DATA '''\n",
    "\n",
    "\n",
    "# rename mag_gtdb user_genome column to match proximeta dataframe\n",
    "mag_gtdb.rename(columns={'user_genome':'host_name'}, inplace=True) \n",
    "\n",
    "# Merge two data frames together\n",
    "proximeta_gtdb = proximeta.merge(mag_gtdb,how='outer',on=[\"host_name\",\"sample_name\",\"sample_depth\"]).replace('', np.nan).fillna(\"NA\")\n",
    "\n",
    "# Replace NA with 'uninfected mag' (MAGs with no associated viruses)\n",
    "proximeta_gtdb['virus_type'] = proximeta_gtdb['virus_type'].replace('NA', 'uninfected mag')\n",
    "\n",
    "# input vmag summary files\n",
    "jv119_vmag_sum = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-119_proximeta/viral_MAGs/viral_mags_summary.tsv'), sep = '\\t')\n",
    "jv121_vmag_sum = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-121_proximeta/viral_MAGs/viral_mags_summary.tsv'), sep = '\\t')\n",
    "jv132_vmag_sum = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-132_proximeta/viral_MAGs/viral_mags_summary.tsv'), sep = '\\t')\n",
    "jv154_vmag_sum = pd.read_csv(op.join(path,'data/proximeta_viral_files/jv-154_proximeta/viral_MAGs/viral_mags_summary.tsv'), sep = '\\t')\n",
    "\n",
    "# create sample_name column\n",
    "jv119_vmag_sum['sample_name'] = 'JV119'\n",
    "jv121_vmag_sum['sample_name'] = 'JV121'\n",
    "jv132_vmag_sum['sample_name'] = 'JV132'\n",
    "jv154_vmag_sum['sample_name'] = 'JV154'\n",
    "\n",
    "# combine all vmag summary dataframes together\n",
    "vmag_sums = pd.concat([jv119_vmag_sum, jv121_vmag_sum, jv132_vmag_sum, jv154_vmag_sum])\n",
    "\n",
    "# split up contig_id column into different columns\n",
    "vmag_sums['virus_name'] = vmag_sums['contig_id'].str.split(\"|\").str[0]\n",
    "vmag_sums['N'] = pd.to_numeric(vmag_sums['contig_id'].str.split(\"|\").str[1].str.replace('N=', ''))\n",
    "vmag_sums['L'] = pd.to_numeric(vmag_sums['contig_id'].str.split(\"|\").str[2].str.replace('L=', ''))\n",
    "\n",
    "# remove contig_id column and turn all empty observations to NA\n",
    "vmag_sums = vmag_sums.drop('contig_id', axis=1).fillna(\"NA\")\n",
    "\n",
    "# merge with proximeta_gtdb\n",
    "proximeta_sum = proximeta_gtdb.merge(vmag_sums,how='outer',on=[\"virus_name\",\"sample_name\"]).replace('', np.nan).fillna(\"NA\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "efd03815-ec8b-4ef7-9916-b43a5e21413b",
   "metadata": {},
   "outputs": [],
   "source": [
    "proximeta_sum.to_csv(op.join(path, 'tables/proximeta_master.csv'))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
